---
title: "Flexibility of interactions depends on habitat suitability"
author: 
  - "E. Fernando Cagua^1^ ([efc29@uclive.ac.nz](mailto:efc29@uclive.ac.nz))"
  - "Audrey Lustig^2^ ([audrey.lustig@canterbury.ac.nz](audrey.lustig@canterbury.ac.nz))"
  - "Jason M. Tylianakis^1^ ([jason.tylianakis@canterbury.ac.nz](jason.tylianakis@canterbury.ac.nz))"
  - "Daniel B. Stouffer^1^ ([daniel.stouffer@canterbury.ac.nz](daniel.stouffer@canterbury.ac.nz))"
bibliography: 
  - bibliography.bib
  - int-bibliography.bib
csl: pnas.csl
output: #word_document
  bookdown::pdf_document2:
    # base_format: rticles::peerj_article
    keep_tex: yes
    number_sections: false
toc: false
fontsize: 11pt
# classoption: a4paper
geometry: 
  - textwidth=33em
  - textheight=48\baselineskip
header-includes:
  - \usepackage{booktabs}
  - \usepackage{setspace}
  - \usepackage{lineno}
  - \usepackage{xr}
  - \usepackage[utf8]{inputenc}
  - \newcommand{\R}[1]{\label{#1}\linelabel{#1}}
  - \newcommand{\lr}[1]{page~\pageref{#1}, line~\lineref{#1}}
  - \externaldocument{supp-info}
  - \usepackage[export]{adjustbox}
  - \usepackage{caption}
editor_options: 
  chunk_output_type: console
---

\linenumbers
\doublespacing
\captionsetup[table]{skip=10pt}

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, 
                      warning = FALSE, dpi = 300, 
                      eval.after = "fig.cap")

library(dplyr)

loadd(abs_wordcount, msc_wordcount, n_displays, n_references, keywords, bibliography)
loadd(interaction_citations)
```
 
^1^ Centre for Integrative Ecology, School of Biological Sciences, University of Canterbury, Private Bag 4800, Christchurch 8041, New Zealand

^2^ Geospatial Research Institute, University of Canterbury, Private Bag 4800, Christchurch 8041, New Zealand

**Running title:** XX

**Keywords:** `r keywords`

**Type of article:** XX

**Number of words:** `r abs_wordcount$n_words_stri` in abstract; `r format(round(msc_wordcount$n_words_stri), big.mark = ",")` in main text.

**Number of displays:** `r n_displays$figures` figures; `r n_displays$tables` tables; `r n_displays$text_boxes` text boxes.

**Number of references:** `r n_references + nrow(interaction_citations)`

**Author for correspondence:** E. Fernando Cagua (+64 20 4026 8153).

**Data accessibility:** Data supporting the results will be accessible in an appropriate data repository after publication. The DOI will be included here.

**Author contributions:** XX

\clearpage

# Abstract

`r paste(drake::readd(abstract), collapse = "")`

\clearpage

# Introduction

Species interactions are known to vary widely across space and time. 
There are multiple examples of species that interact tightly in an ecological community or a particular season but not in another. 
It is crucial to understand why and how this turnover of interactions occur because the network of interactions of an ecological community is known to underpin ecosystem function and stability. 
Despite much progress in the last decades, we are still unable to predict species interactions, partially because we still have not accurately identified the factors that drive species to interact under certain conditions and not under another. 

There are two main processes involved in whether two species in a community interact or not.
The first is an "Eltonian" niche process that is underpinned by the traits of the interacting species.
Under this rationale, when, for example, a deep-rooted co-evolutionary history leads to optimally matched traits, the interacting species can obtain a substantial benefit from this interaction. 
Contrastingly, we would expect a lower per-capita benefit from interactions with low trait matching.
The second is a neutral process underpinned by the encounter rate of the interacting species. 
All else being equal, when the encounter rate of interacting species is high there is a higher probability to interact than if the rate were smaller, for example due to species rarity. 

Perhaps, because it operates at scales larger than those of the Eltonian niche concept, the abiotic environment is one of the factors that is commonly overlooked by community ecologists when explaining interaction turnover [but see @gravel_bringing_2018]. 
The most evident—and best-studied—way through which the environment can affect the structure of ecological networks is by way of patterns of community composition. 
At the heart of biogeography is the idea that the abiotic environment can determine whether a species can have a positive growth-rate when rare in a particular location.
For two species to interact, they first need to co-occur in the same community, which can only happen if the environmental conditions of the community fall within the "Grinellian" niche of both species. 
However, community composition is only one part of the story as the abiotic environment can also influence the interactions directly [@tylianakis_ecological_2017]. 

When looked from an Eltonian-niche lens, one way the environment can influence wether two species interact or not is based on, what we call, stress-induced hypotheses of interaction turnover. 
These hypotheses are rooted in the co-evolutionary processes that determine the quality of an interaction partner. 
Environmental induced stress can 

Another way

In a pollination context, it can for plant–pollinator interactions (Berg et al., 2010; Elzinga et al., 2007)
. 

Second, and oare more subtle, mechanism through which enviroment can determine interactions is
This is so because , species distributions are strongly determined 
At the basis of species distrFormally, the fundamental "Grinellian" niche of a species is determined by the set of environments where a species has a positive groth rate when rare. 
The environement, 
From a "Grinellian" niche perspect
Framed from a niche perspective, 

It is generally agreed that two species wether two species interact there is a neutral process

HYPOTHESIS: many trophic and mutualistic interactions are relatively generalized and potentially opportunistic (Pimm et al. 1991). Thus, we may expect that ecological networks will comprise a mixture of interactions that persist over the long-term owing to trait matching of partners through coevolution (Go ́ mez et al. 2010, Guimaraes et al. 2011) and those that occur opportunistically, perhaps with inferior long-term fitness benefits (Aizen et al. 2016).

It is generally accepted that the species niche is the location in a multidimentional space defined by environmental conditions that allows a species to have a positive growth rate when rare (Godsoe). 
There are two dimensions of a species niche, the Grinellian niche which focuses on the 

<!-- Interactions are selected by species depending on the relative benefit they provide. The assumption goes that species with a large number of interactions are benefit relatively little from each of them, and species with a small number benefit largely with them, this is they focus in more optimal pollinators. -->

<!-- AND  -->

<!-- We know that interactions are influenced by the environment.  -->

<!-- BUT -->

<!-- We don't know how exactly wether the enviroment influences the patterns of generalisation/specialization.  -->

<!-- One option is that when species are in detrimental environmental conditions focus on most optimal partners and as such interact with a smaller number of species.  -->
<!-- An alternative hypotheisis is that when species are in favourable environmental conditions they can afford to be more picky about their interactions and as such avoid those of less value, while individuals in unfavourable conditions must get rewards from whatever they can get.  -->

<!-- THEREFORE -->

<!-- We used a global dataset on plant pollinator interactions, species occurrence, and bioclimatic data to test these hypotheses.  -->
<!-- If the first option is true we should see a negative relationship between environmental sutability and the relative degree of a species.  -->
<!-- If the second option is true we should see the opposite, a positive relationship between environmental suitability and the relative degree of a species.  -->




<!-- **Is my central topic turnover of interactions**? If yes then I should quanitify turnover accross high and low suitability areas.  -->

<!-- **Continuation: compare the suitability of the species that species do not interact with vs those that interact (when a possible interaction occurs but is not realised)?** -->

<!-- **Continuation: average suitability of all species. If avg. competitors suitability is low, species, facilitation more common and resource sharing should increase. If avg. suitability is high, the opposite should happen?** -->

<!-- **Continuation: Should I use relative suitability instead of absolute? Is not total suitability what matters but that relative to other species?**  -->

<!-- **Is smaller degree a smaller Eltonian niche?** Should I frame this as the relationship between Grinellian and Eltonian. A better position in the grinellian niche allows for a larger Realised Eltonian niche. A problem of stating the hypothesis in terms of niches is that there are many definitions of niche and people is picky -->

<!-- *Previous state of the world: We know interactions change with the environment but we don't know beyond co-occurrence patterns* -->

<!-- Notes: -->

<!-- **From donoso et al 2017: ** Intro: It is generally agreed that both neutral and niche-driven processes contribute to the occurrence of pairwise mutualistic interactions and network structure. That is, whether or not two species interact can be thought of as the result of two sequential processes. First, there is a stochastic (neutral) process driven by the local abundances of interacting species, such that abundant species are more likely to encounter each other than are rare ones. Secondly, there is a deterministic process driven by a matching of traits (e.g., phenology, morphology, behavior, etc.) that finally permits the interaction between co- occurring species.  -->

<!-- Intro: Mutualistic interactions are fundamentally dynamic, such that there is a turnover of species and individuals in space and time (Carnicer et al., 2009; Laliberté and Tylianakis, 2010; Hagen et al., 2012).  -->

<!-- I would then add a thrird one that is dictated by the rewards provided by an interaction. Maybe? -->

<!-- Conclusion: Overall we found that the assemblage of pairwise interactions depended on the neutral effects of interacting species abundances (the more abundant the species of fruit and frugivore, the more frequent their pairwise interaction) and on the local context of ecological neighborhoods. -->

<!-- **From Boulangeat et al 2012:** Intro: These drivers of species distributions may act at different spatial scales. For instance, frequent consideration is given to abiotic variables acting at large spatial scales. On the other hand, biotic interactions and consumer-resource dynamics are assumed to occur at smaller spatial scales. -->

<!-- Conclusion: Biotic inetractions are def important at determining abundace. They observed positive effects of biotic interactions at colder climates (low suitability for plants) -- suggest facilitation. and negative effects of interactions at warmer climates (high suitability) suggesting competition. *Should I calculate the suitability of other species and test this?* -->

<!-- **From Dormann et al 2018:** Intro: Sometimes, an organism will be little affected by com- petitors, mutualists, pathogens or predators; a mature tree may appear to grow unfettered by its neighbours for decades. In other cases, an individual may struggle to survive the pressure of its community during every hour of its life; for example, a fish preying and escaping predators in a small river. Biotic interactions such as these can have major effects on local processes and population dynamics, and some, but not neces- sarily all, scale up to co-determining the range of a species (reviewed, e.g., by Wisz et al., 2013). -->

<!-- Most BI-SDMs assume that biotic interactions are constant in space and time. It is more likely, however, that interaction strengths are modulated by the biotic and abiotic environment. Experimental studies showed that rates of biological interactions depend on environmental conditions: rates of herbivory are temperature dependent (Burnside, Erhardt, Hammond, & Brown, 2014); mycorrhizal symbioses change with resource availability (Cox, Barsoum, Lilleskov, & Bidartondo, 2010; Johnson, Wilson, Bowker, Wilson, & Miller, 2010); and plant–plant interactions can shift from competitive to facilitative along a stress gra- dient (Bertness & Callaway, 1994). Additionally, environmental change can modulate biotic interactions via phenological mismatches, such as for plant–pollinator interactions (Berg et al., 2010; Elzinga et al., 2007). In these conditions, the model must also allow for the interplay of environmental and biotic interactions (Clark et al., 2011), which has only recently started to be explored in BI-SDMs (Tikhonov, Abrego, Dunson, & Ovaskainen, 2017). -->

<!-- **From Gravel et al. 2018**: We apply this framework to host–parasite interactions across Europe and ind that two aspects of the environment (temperature and precipitation) exert a strong imprint on species co-occurrence, but not on species interactions. Even where species co-occur, interaction proves to be stochastic rather than deterministic, adding to variation in realized network structure. *Bingo: I show it's not completely stochastic and is likely to be determined by the suitability hehe* -- for real ( They test prob. of two co-occurring species to interact does not depend on environment. I test single species thingy). Should I do that? suit sp.1 and suit sp. 2? It;s not about the environment, but about how the environment affect the species involved. Buya! -->

<!-- There is, however, ample evidence that interaction networks vary in space and time (Laliberté and Tylianakis 2010, Poisot et al. 2012, 2016c, Schleuning et al. 2012, Albouy et al. 2014, Trøjelsgaard et al. 2015), even though there is no common framework with which to generalize these results. -->

<!-- he occurrence and intensity of interactions could vary because of the environment, species abundance, and higher- order ecological interactions (Poisot et al. 2015). *Well do I show that could vary because of more stuff?* -->

<!-- Variation in community composition can be independent of variation of ecological interactions, suggesting that species and inter- ac tion distribution may well respond to diferent drivers (Poisot et al. 2012). -->

<!-- Ecological interactions could also vary with the environment. Although it is rare to see a conditional representation of pairwise ecological interactions, experimental studies have frequently revealed interactions to be sensitive to the environment. For instance, McKinnon et al. (2010) showed that predation risks of shorebirds vary at the continental scale, decreasing from the south to the north. It is also common to see increasing top-down control with temperature (Shurin et al. 2012, Gray et al. 2016). Efects of the environment on interactions (Gibert and DeLong 2014) also propagate up the community and inluence net- work structure (Tylianakis et al. 2007, Petchey et al. 2010, Woodward et al. 2010). -->

<!-- In terms of species interacting with each other, our analysis suggests that the environment leaves a detectable imprint on species co-occurrence, but only a slight mark on the occurrence of realized links among species in a speciic place: the prob- ability of inding a given combination of species at a higher and a lower trophic level at the same site was clearly afected by the environment, whereas the probability of observing an interaction between the two was, given co-occurrence, not signiicantly so.  -->

<!-- This does not imply that species will always interact when they meet – **although this is a basic assumption in most documented metawebs to date** (Havens 1992, Wood et al. 2015).  -->

<!-- Evidence that the structure of ecological networks does vary across habitats (Tylianakis et al. 2007), over environmental gradients (Lurgi et al. 2012) and in time (Simanonok and Burkle 2014) is accumulating rapidly.  -->

<!-- **Devictor 2010:** I can use degree to quantify realised eltonian niche -->

<!-- **Godsoe 2017**: Biogeographers and community ecologists have long debated the importance of biotic interactions (see Glossary) for species distributions and range limits [1–5].*But seldom the other way arround?* -->

<!-- **Tylianakis 2017**: Interactions can change because of environment because environment and habitat can change encounter rate.  -->

<!-- Specialist species (that lack flexibility in their choice of interaction partner) should theoretically be most vulnerable to phenological decoupling. However, one study of pollinator communities along an elevational gradient found that apparently specialized pollinators were facultatively more generalist than they appeared and could adapt their visitation patterns across sites as much as generalists to avoid phenological asynchrony (Benadi et al. 2014). Similarly, duration of the fruiting season has been shown to predict the degree and strength of seed-dispersal interactions (Gonza ́ lez- Castro et al. 2012), and a long-term study found no difference between specialist and generalist plant species in their interannual fluctuations in phenology and pollinator richness (Petanidou et al. 2014). Although these findings suggest that the facultative generalism of pollination mutualisms may confer robustness to these networks in the face of altered phenology (see also discussion on rewiring above), this robustness may depend on the climate context and the diversity and specialization within the network (Hoiss et al. 2015). Moreover, less-generalized interactions such as those between parasites and hosts may show less flexibility. -->

<!-- many trophic and mutualistic interactions are relatively generalized and potentially opportunistic (Pimm et al. 1991). Thus, we may expect that ecological networks will comprise a mixture of interactions that persist over the long-term owing to trait matching of partners through coevolution (Go ́ mez et al. 2010, Guimaraes et al. 2011) and those that occur opportunistically, perhaps with inferior long-term fitness benefits (Aizen et al. 2016). Increases in the frequency of the former will strengthen any pattern of phylogenetic congruence, whereas increases in the latter will erode it. -->

<!-- For example, environmental stressors could cause species to narrow their trophic breadth by increasing the search time to find alternative partners (Gauld 1987) or by imposing limits on the quality of interaction partners that can be used (i.e., using only resources to which a species is best adapted). Phylogenetic congruence becomes stronger in parasitoid–host networks with increasing temperature (Lavandero & Tylianakis 2013) and in plantation versus native forests (Peralta et al. 2015), and such patterns appear consistent with a hy- pothesis of stress-induced reduction in trophic (phylogenetic) breadth (in the case of Lavandero & Tylianakis 2013, this occurred on genotype relatedness rather than between-species phylogenies). However, evidence for the hypothesized mechanism is still lacking, as experiments have revealed no change in the proportionate use of good or bad hosts when parasitoids are nutritionally stressed (Hopper et al. 2013) or even decreased selectivity toward high-quality hosts under high levels of nutritional stress (Crespo et al. 2015). -->

<!-- In addition to seasonal patterns, interaction turnover can be driven by environmental fac- tors, such as community changes during drought years (Alarco ́n et al. 2008) or along land use or landscape intensity gradients (Gagic et al. 2012, Laliberte ́ & Tylianakis 2010). The impact of environmental conditions on interaction turnover can, in turn, be constrained by species traits such as feeding morphology (Carnicer et al. 2009) or body size (Laliberte ́ & Tylianakis 2010). To understand this balance, species interactions can be thought of as having both a niche-based or trait-based component and a random or neutral component determined by encounter probability (Bartomeus et al. 2016, Vazquez et al. 2009). Any environmental driver that sufficiently increases encounter probability (see Section 2.2) for all species could reduce the neutral barrier to inter- actions and allow species to consistently select their interaction partner with optimally matched traits. This enhanced expression of species preferences would result in reduced spatial or temporal variability because the same interactions would consistently be realized (i.e., network homogeniza- tion; Laliberte ́ & Tylianakis 2010). Reduction of the neutral component of interactions could also lead to exaggeration of patterns of trait matching and phylogenetic congruence (Carnicer et al. 2009) (see Section 2.3). Alternatively, a driver that increases the abundance of a subset of species could result in their overrepresentation in interactions by allowing the neutral component of inter- actions to be overexpressed relative to the trait-based component (Carnicer et al. 2009). Because individual species may differ in the extent to which they engage in neutral interactions, changes to the relative abundance of species can alter network architecture. For example, exotic birds can be less selective in the fruit species on which they feed (i.e., feeding more neutrally on species in proportion to their abundance), and this low selectivity can allow them to fill the gap left by rare or extinct native birds and maintain high network nestedness and connectance (Garcıa et al. 2014). -->

<!-- **Peralta 2015:** Plantation forests (as opposed to native) habitat may impose stresses on parasitoids that constrain them to attack only host species for which they are best adapted. -> Consistent with more stress less generalism.  -->

<!-- Native herbivore–parasitoid combinations whose interac- tions contributed the most to the pattern of phylogenetic congruence were more abundant in plantations than in native forests. This stronger co-evolutionary signal in plantation forests suggests that parasitoids in disturbed habitats may only be able to utilize hosts that they have best evolved to attack. Analogously, climate warming has been shown to reduce the niche breadth plasticity of par- asitoids (Lavandero & Tylianakis 2013), suggesting that different environmental changes could reduce the ability of parasitoid communities to suppress a broad range of host species and genotypes in natural systems and biologi- cal control programmes. -->

# Methods

## Data aquisition

### Species interactions

```{r}
loadd(config)
loadd(int_metadata)

n_published_studies <- int_metadata$reference %>%
  unique() %>%
  stringr::str_detect(pattern = "unpubl", negate = TRUE) %>%
  sum()

all_int_cites <- paste0("@", interaction_citations$BIBTEXKEY) %>%
  glue::glue_collapse(sep = "; ") %>%
  paste0("[", ., "]")

int_citations_year_range <- range(interaction_citations$YEAR)
```

```{r scientific-names-cleaning-info}
loadd(checked_sp_names, spp, manual_name_corrections, problematic_networks)

n_sp_names <- n_distinct(spp$sp_name) 

n_sp_names_subsp_level <- filter(spp, is_subspecies) %$%
  n_distinct(sp_name)

n_sp_names_sp_level_no_subsp <- filter(spp, !sp_unidentified, !is_subspecies) %$% 
  n_distinct(sp_name)

n_sp_names_sp_level <- filter(spp, !sp_unidentified) %$% 
  n_distinct(sp_name)

n_sp_names_gen_level <- filter(spp, sp_unidentified, !gen_unidentified) %$% 
  n_distinct(sp_name)

n_no_tax_info <- filter(spp, gen_unidentified) %$% n_distinct(sp_name) 

n_sp_correct_in_data <- checked_sp_names %>%
  filter(good_queried_sp_name) %$%
  n_distinct(queried_sp_name)

n_sp_corrected_typos <- checked_sp_names %>%
  filter(corrected_typo) %$% 
  n_distinct(queried_sp_name)

sp_with_synonyms <- checked_sp_names %>%
  filter(alt_sp_name_found) 

n_sp_with_synonyms <- sp_with_synonyms %$%
  n_distinct(queried_sp_name)

sp_not_found <- checked_sp_names %>%
  filter(!good_queried_sp_name) %>% 
  filter(!alt_sp_name_found, !corrected_typo) 

n_sp_not_found <- sp_not_found %$%
  n_distinct(queried_sp_name)

n_sp_manual_corrections <- manual_name_corrections %>%
  filter(!is.na(sp_name)) %$%
  n_distinct(queried_sp_name)

study_per_location <- int_metadata %>%
  distinct(loc_id, reference)

n_names_in_multiple_references_orig <- spp %>%
  left_join(study_per_location, by = "loc_id") %>%
  group_by(sp_name) %>%
  summarise(n_net = n_distinct(loc_id), 
            n_ref = n_distinct(reference)) %>%
  filter(n_ref > 1) %$%
  n_distinct(sp_name)
```

We retrieved plant-pollinator networks from the Web of Life database [@fortuna_web_2014]. 
This database contains datasets originating from `r length(interaction_citations$BIBTEXKEY)` studies published in the primary literature between `r int_citations_year_range[1]` and `r int_citations_year_range[2]` `r all_int_cites`.
All together, these data sources contained `r format(n_sp_names, big.mark = ",")` unique organism names; `r format(n_names_in_multiple_references_orig, big.mark = ",")` of them in more than one data source.
Because assessing how habitat suitability affects interaction partners of plants and pollinators requires linking of species across different datasets, it was important to ensure that the names used to identify species were as accurate as possible.
From the total number of organisms,`r format(n_sp_names_subsp_level, big.mark = ",")` were identified to the subspecies or variety level, `r format(n_sp_names_sp_level_no_subsp, big.mark = ",")` to the species level, `r format(n_sp_names_gen_level, big.mark = ",")` to the genus level, and no taxonomic information was avaliable for the remaining `r format(n_no_tax_info, big.mark = ",")`. 
As the species level was the most common taxonomic rank available in our interaction datasets, in all further analysis, we group together subspecies or varieties within a same species. 

We were able to confirm the validity of `r format(n_sp_correct_in_data, big.mark = ",")` of the cannonical names used to identify organisms (roughly `r scales::percent(n_sp_correct_in_data/n_sp_names_sp_level, 1)`). 
We assessed the validity of a cannonical name by querying the Global Names Resolver database (https://resolver.globalnames.org) which includes data from `r dplyr::n_distinct(taxize::gnr_datasources()$title)` taxonomic sources. 
Specifically, we used the function `gnr_resolve` from the R package `taxize` `r as.character(packageVersion('taxize'))` [@chamberlain_taxize_2013; @chamberlain_taxize_2019].

From the remaining `r format(n_sp_names_sp_level - n_sp_correct_in_data, big.mark = ",")` names we were unable to validate, we were able to identify and correct `r format(n_sp_corrected_typos, big.mark = ",")` spelling mistakes. 
These spelling mistakes were corrected automatically by fuzzy matching the canonical names in our data sources with tgose in the Global Names Resover database. 
In rare occasions, the fuzzy matching algorithm can suggest a cannonical name that has similar spelling but that correspond to an organism in a completely different taxonomic group, often in a different kingdom. 
We, therefore, checked the taxonomic hierarchy of suggested names to confirm that it corresponded to our expectations. 
We retrieved all taxonomic hierarchies from the National Center for Biotechnology Information taxonomic database (https://www.ncbi.nlm.nih.gov/taxonomy). 

Subsequently, we checked for possible synonyms of the canonnical names in our data sources. 
Using data from the Integrated Taxonomic Information System database (http://www.itis.gov), we found synonyms and alternative names for `r format(n_sp_with_synonyms, big.mark = ",")` of those in our data sources.
Finding these alternative names was important due to two reasons. 
First, because we wanted to be able to identify the cases in which the same organism might have been recorded with different cannonical names in different data sources. This can ocurr not only with synonyms but also when the cannonical name has been changed or when there are widely used ortographic variants.
Second, because retrieving occurrence data is often only possible using the latest accepted/valid name for a particular species.

All together, from the `r format(n_sp_names_sp_level - n_sp_correct_in_data, big.mark = ",")` names we were unable to validate, it was not possible to automatically correct `r format(n_sp_not_found, big.mark = ",")` of them. 
We then manually consulted multiple online databases, chiefly Wikispecies (https://species.wikimedia.org/), and looked for cannonical names that resembled the unvalidated names and matched the geographic and taxonomic spectations. 
We were able to further correct `r n_sp_manual_corrections` names in this fashion.
Most manual corrections were of species names that have been abbreviated or had more than two spelling mistakes. 
A complete list of manual name corrections can bee seen in Table \@ref(tab:tab-manual-name-corrections).

```{r}
loadd(recoded_interactions, clean_interactions)

n_sp_ids <- recoded_interactions %>%
  select(-loc_id) %>%
  purrr::map(n_distinct) %>%
  unlist() %>% sum()

n_names_in_multiple_references_after <- recoded_interactions %>%
  tidyr::gather(key = "guild", "sp_id", pla_id, ani_id) %>%
  left_join(study_per_location, by = "loc_id") %>%
  group_by(sp_id) %>%
  summarise(n_ref = n_distinct(reference)) %>%
  filter(n_ref > 1) %$%
  n_distinct(sp_id)

stats_clean_int <- clean_interactions %>%
  purrr::map(n_distinct)
```

This cleaning process allowed us to match further `r n_sp_names - n_sp_ids` names across data sources and, by doing so, identify another `r n_names_in_multiple_references_after - n_names_in_multiple_references_orig` species that were present in more than one study. 
It also allowed us to identify `r numbers2words(length(problematic_networks))` problematic networks in which some names were included as both plants and pollinators. 
These networks were removed from further analysis. 
Sometimes interaction data was recorded at multiple points in time within or across our data sources. 
When this was the case, we combined them into one single interaction network per location. 
The cleaning process resulted on a total of `r format(stats_clean_int$pla_id, big.mark = ",")` plants and `r format(stats_clean_int$ani_id, big.mark = ",")` pollinator species distributed across `r format(stats_clean_int$loc_id)` locations arround the globe (Figure \@ref(fig:fig-worldmap), \@ref(fig:fig-distribution-species-mult-locations)).

```{r fig-worldmap, fig.width = width('single'), fig.cap ="Worldwide distribution of pollination communities included in this study"}
readd(fig_worldmap)
```

```{r fig-distribution-species-mult-locations, fig.width = width('single'), fig.height = 2.2, fig.cap = fig_dist_species_multiple_locations_legend}

most_popular_species <- readd(fig_dist_species_multiple_locations_data) %>%
  filter(n_locations == max(n_locations)) %>%
  arrange(guild) 

fig_dist_species_multiple_locations_legend <- 
  paste0("Frequency distribution of the number of locations in which a species is present. The most common pollinator species was *", 
       most_popular_species$sp_name[1], 
       "*, which was sampled on ", 
       most_popular_species$n_locations[1], 
       " locations, while the most common plant species was *", 
       most_popular_species$sp_name[2], 
       "*, which was sampled on ", 
       most_popular_species$n_locations[2], 
       " locations.")

readd(fig_dist_species_multiple_locations)
```

### Species suitability

```{r}
loadd(n_occurrences, n_cleaned_occurrences, n_dirty_occurrences, gbif_key_groups, org_ids, n_thinned_occurrences, complete_climate_cells, complete_filled_cells_1, complete_filled_cells_2)

get_n_species <- . %>%
  dplyr::inner_join(gbif_key_groups, by = "taxonKey") %>%
  dplyr::inner_join(org_ids, by = c("key_id" = "sp_key_id")) %$%
  dplyr::n_distinct(org_id)

n_species_with_occurrences <- get_n_species(n_occurrences)
n_species_with_cleaned_occurrences <- get_n_species(n_cleaned_occurrences) 
n_species_with_dirty_occurrences <- get_n_species(n_dirty_occurrences)
percent_thinned <- 1 - sum(n_thinned_occurrences$N) / sum(n_cleaned_occurrences$N)
percent_filled_round_one <- 1-complete_filled_cells_1[1]/complete_climate_cells[1]
```

Our next step was to determine the suitability of the species that were present in at least `r numbers2words(readd(minimum_spp_locations))` communities.
To do that we first fit a climatic envelope for each  of them which requires information about the climate and habitat at the locations where the species occurrs. 
We retrieved climatic data from WorldClim V2.0 which includes 19 bioclimatic variables commonly used in species distribution modelling [@fick_worldclim_2017]. 
These bioclimatic variables were complemented with 18 of bioclimatic and topographic variables available in Envirem [@title_envirem_2018]. 
This additional set of variables have been shown to be relevant to ecological or physiological processes and as such have the potential to improve species distribition models.
All climatic data was obtained as rasters composed by cells of 2.5 arc-minutes. 
This resolution provides a reasonable match to the locational accuracy of the species occurrences found in GBIF, particularly those that come preserved specimens in museum collections. 
Furthermore, we obtained habitat data from The Nature Conservation's terrestrial ecoregions of the world [@olson_global_2002].

We retrieved occurrences from the Global Biodiversity Information Facility (GBIF; https://www.gbif.org) using the R package `rgbif` `r as.character(packageVersion('taxize'))` [@chamberlain_r_2017; @chamberlain_rgbif_2019].
For each species, we downloaded occurrences for which the coordinates of the observation were available and that had no known geospatial issue in the GBIF database.
Roughly, we downloaded `r round(sum(n_occurrences$N)/1000000, digits = 1)` million occurrences for `r format(n_species_with_occurrences, big.mark = ",")` of the species of interest. 
The occurrences obtained contain observations of mixed quality, so we applied a series of simple filters to remove those of lower quality. 
\R{rationale-occurrence-filtering}Specifically, we removed all occurrences with (*i*) a coordinate uncertainity larger than 100km; (*ii*) those recorded prior to 1945 (ref? or why?); (*iii*) those in which the number of individuals recorded was either zero or more than 100 as such large counts are often erroneous (ref? or why?); and (*iv*) those occurrences in which the "basis of record" was not a human observation or a preserved specimen (ref? why?). 
We then used the R package `CoordinateCleaner` `r as.character(packageVersion('taxize'))` [@zizka_coordinatecleaner_2019] and land mass and country data from Natural Earth (https://www.naturalearthdata.com) with a 1:10,000,000 scale to further identify and remove problematic occurrences. 
We removed occurrences in which their coordinates (*v*) fell outside the borders of the country in which they were recorded; (*vi*) those arround a the country capital or the centroid of the country and province centroids; (*vii*) those arond a biodiversity institution; and (*viii*) those located within oceans. 
Thorugh this cleaning process, we removed with `r round(sum(n_dirty_occurrences$N)/1000000, digits = 1)` million occurrences distributed accross `r n_species_with_dirty_occurrences` species. 
We added the occurrences from our plant-pollinator networks to the cleaned occurrences retrieved from GBIF.

Next, we overlapped the occurrence and our environmental rasters such that each occurrence location is characterized by a vector with information of our 37 bioclimatic and topographic variables . 
If a species had more than one occurrence records within one of the cells of the bioclimatic raster we only included one of the occurrence records. 
We did this to avoid giving more weight to areas with a high number of occurrences, a common scenario in occurrrence records collected opportunistically as the ones we use here. 
Overall we removed `r scales::percent(percent_thinned)` of the occurrences which resulted in a total of `r round(sum(n_thinned_occurrences$N)/1000000, digits = 1)` million occurrences used in our niche analysis. 
A common issue of terrestrial bioclimatic datasets is that the boundaries of the cells with information do not exactly match the land mass boundaries. 
The result of this missmatch is that at least one of the environmental variables was not available for `r format(complete_climate_cells[1], big.mark = ",")` of the raster cells with occurrences (`r scales::percent(complete_climate_cells[1]/sum(complete_climate_cells), accuracy = 0.1)` of the total). 
As expected a vast majority of these cells were close to the shore. 
To address this issue, calculated average value of environmental variables within a `r config$climate_buffer/1000`km buffer of the center of the cell where the variable is missing, and used it to approximate the value of the variable in that cell. 
In this fashion we were able to fill environmental variables for `r scales::percent(percent_filled_round_one, accuracy = 0.1)` of the cells were they were missing. 
To fill the remaining `r format(complete_filled_cells_1[1], big.mark= ",")` cells, we repeated the aforementioned procedure but using a `r config$climate_buffer*2/1000`km buffer. 
We removed from further analyis occurrences located within the `r complete_filled_cells_2[1]` cells for which we were unable to fill environmental variables. 

Our main objective is to investigate wether the suitability of the environment was to obtain the suitability of the environment at the locations for which we have community data. 
This involves two steps. 
First, we determine the species' grinellian niche, the range of environmental conditions that are suitable for a positive growth rate. 
We assume that the suitability of a particular set of environmental conditions (a point in the environmental space) can be inferred from the density of occurrences under these conditions. 
This is, areas of the niche that have a higher density of occurrences have higher suitability and can be interpreted as the core of the species distribution in environental space. 
Contrastingly, areas with a lower density are comparatively less suitable and represent the fringes of the species distribution. 
The second step is to determine the location of our plant-pollinator communities (where the species is present) in the environmental space and calculate its corresponding environmental suitability.

To determine the environmental space we use the first two components from a principal component analysis of the 37 bioclimatic variables associated with the species occurrences. 
Specifically we use the `dudi.pca` function from the R package `ade4` `r as.character(packageVersion('ade4'))` [@dray_ade4_2007] and center and scale all bioclimatic variables to have a mean of 0 and a unit standard deviation. 
We then determine the position of species occurrences in the environmental space and estimate their bivariate density distribution using the kernel method. 
The main advantage of using the kernel method is that it reduces the sensitivity of the niche to sampling effort and the resolution chosen for the environmental space grid [@broennimann_measuring_2012]. 
To calculate the kernel we use `ecospat.grid.clim.dyn` from the R package `ecospat` `r as.character(packageVersion('ecospat'))` [@ecospat_citation] with a grid resolution of 200.
We then normalise the occurrence density distribution such that our suitability ranges beetween 1 for the grids with environmental conditions in which the species was most commonly observed and 0 for the grids in which it was never observed.  
We then determine the location in environmental space of the plant-pollinator communities where the species is present using the function `suprow` from `ade4` and calculate its corresponding suitability using the R package `raster` `r as.character(packageVersion('raster'))` [@raster_citation]. 

```{r sensitivity-analysis-calculations}
loadd(sensitivity_species_name)
loadd(n_enough_occurrences)
n_occ_sensitivity_species <- n_thinned_occurrences %>%
  dplyr::filter(org_id == sensitivity_species_name$org_id[1]) %$% N

n_spp_not_enough_occ <- nrow(n_thinned_occurrences) - nrow(n_enough_occurrences)
```

We used a sensitivity analysis to determine the minimum number of occurrences that are necessary to have robust environmental suitability values in our communities. 
For that we used the species with most occurrences available, *`r sensitivity_species_name$canonicalName`*, and calculated the mean absolute error of the suitability values obtained with `r numbers2words(config$n_subsamples)` subsamples from the `r format(n_occ_sensitivity_species, big.mark  =",")`  occurrences available from GBIF. 

## Model

We then used a two sets of bayesian multilevel models to evaluate the impact of environmental suitability on the size of the realised Eltonian niche. 
We measure the size of the Eltonian niche using the normalised degree of species, this is, the number of species it interacts with given the number of possible partners. 
In the first set of models, we use a *naive* version of the normalised degree [@martin_gonzalez_centrality_2010] in which the number of possible partners is defined by the number of species in the opposite guild.
This definition is the one commonly used in the ecological network literature. 
In the second set of models, we use an *informed* version of the normalised degree in which the number of possible partners is defined by the number of species in the community with which the focal species is known to interact in any other location in our dataset.
Measuring the size of the Eltonian niche using the informed version of normalised degree is a particularly stringent test of the effect of environmental suitability as for many species populations in our dataset, the number of possible interacting species is the same or very similar to the number of species it actually interacts with. 

In all models normalised degree was modelled using a logit link function and a binomial distribution in which the number of species interacts with is the number of successes and the number of possible partners is the number of trials. 
We are aware that wether species intreacts or not is not a bernoulli process as species interactions are not independent from each other, however a binomial distribution allow us to indirectly account for the differences in species richness across communities. 
Importantly, results are qualitatively the similar when we model species degree directly using a Poisson distribution and a logarithmic link function and include the number of possible partners as predictors (Supp info).

Within each model set we evaluate four candidate models in order to evaluate the relative importance of suitability. 
A first model, our baseline model, included four population-level predictors and two grouping levels, species and the community. 
The population-level predictors, commonly called fixed effects, were the environmental suitability, the species guild, the species generalism, and it's environmental niche breadth.
The environmental suitability of the species in that particular community, is an index defined betwen zero and one in which a value of zero indicates that the species has never been observed in this environment while a value of one indicates that the species is most commonly observed in that environment. 
We were particularly intrested on understanding wether the effect of species suitability is conditional on the species guild (plant or pollinator), we therefore included guild and its interaction with suitability in the model. 
To estimate a species overal level of generalism we included among the predictors the magnitude of the total number of interacting species across communities in our dataset. 
We included the number of communities where the species was present as a rough estimate of a species environmental niche breadth. 
We allowed the intercept of degree and slope of the suitability-degree relationship to vary among species, this allowed us to investigate the extent to which suitability is a population or a group level effect. 
Finally, to account for unmeasured differences between communities, like sampling effort, sampling method, or diversity, we calculated an intercept for each community in our study.

The second, third, and fourth models within each model set were a reduction of the first. 
In the second model we removed the species generalism as a predictor, in the third, we removed the environmental niche breadth, and in the fourth model we removed excluded the effect of environmental suitability (both at the population and the species level). 
To facilitate model interpretation and convergence, we scaled all continuous variables to have a mean of zero and a unit standard deviation. 

All models were fitted under a bayesian framework using the R package `brms` `r as.character(packageVersion('brms'))` [@burkner_brms_2017; @burkner_advanced_2018] as an interface for Stan [@carpenter_stan_2017]. 
All models were fitted using four Markov chains of 4,000 iterations each; half of the iterations were used for warmup.
We used weakly informative priors for all model parameters. 
Specifically we used normal priors of mean zero and standard deviation ten for the population level effects and the intercepts, a half-Cauchy prior with a location of zero and a scale of two for the standard deviations, and, when applicable, an LKJ-correlation prior with parameter \(\zeta = 1\) for the correlation matrix between group-level parameters. 
To minimise the potential effect of influential observations we used leave-one-out cross-valiadtion to estimate their expected log predictive density and compare the out-of sample performance of the different models. 
We also calculate the Wanatabe-Akaike information criterion (WAIC) for comparison. 

# Results

After performing our sensitivity analysis, we found that we need roughly `r readd(min_occurrences_factor)` independent occurrences for each community for which we calculated a suitability value in order to obtain a mean absolute error below `r config$min_suitability_error` (Fig. \@ref(fig:fig-sensitivity-analysis)). 
We therefore removed from further analyses `r n_spp_not_enough_occ` species for which we did not have enough occurrences to obtain robust estimates.

```{r fig-sensitivity-analysis, fig.width = width('single'), fig.height = 2.2, fig.cap = fig_sensitivity_analysis_legend}

fig_sensitivity_analysis_legend <- 
  paste0("Sensitivity analysis of environmental suitability error. ",
         "The number of independent occurrences retrieved from GBIF is inversely related to the error of environmental suitability for our plant-pollinator networks.", 
         "The sensitivity analysis was performed by subsampling occurrences of", 
         " *", sensitivity_species_name$canonicalName, "* ", 
         "the species in our dataset with the largest number of occurrences in GBIF, which was recorded in two of our communities.")

readd(fig_sensitivity_analysis)
```

```{r model-results-calculations}
loadd(bayesian_r2_baseline)
loadd(parameter_posterior_summary)
loadd(model_ranking)

random_cor <- parameter_posterior_summary %>%
  dplyr::filter(parameter == "cor_org_id__Intercept__scaled_suitability")

sd_loc_id <- parameter_posterior_summary %>% 
  dplyr::filter(stringr::str_detect(parameter, "sd_loc_id__Intercept"))

sd_org_id <- parameter_posterior_summary %>% 
  dplyr::filter(stringr::str_detect(parameter, "sd_org_id__Intercept"))

sd_org_id_suitability <- parameter_posterior_summary %>% 
  dplyr::filter(stringr::str_detect(parameter, "sd_org_id__scaled_suitability"))

waic_comp_suitability <- model_ranking$waic$formula_no_suitability$pointwise - model_ranking$waic$formula_base$pointwise 

waic_diff_suitability <- sum(waic_comp_suitability[, "waic"])

waic_se_diff_suitability <- sqrt(nrow(waic_comp_suitability) * var(waic_comp_suitability[, "waic"]))
```

We found that our models performed relativelly well.
The bayesian R-squared for our baseline model was `r round(bayesian_r2_baseline[1], digits = 2)`, which indicates our models were able to capture a large proporton of the variability on the data. 
Overall we found only weak evidence of a relationship between the environmental niche size and the normalised degree (Figure \@ref(fig:fig-conditional-effects)a). 
This relationship was stronger for plants than for animals.
Similarly environmental suitability does not show a consistent pattern accross species. 
Indeed, when looking at the population level effects, suitability virtually no relationship with normalised degree, neither for plants or pollinators (Figure \@ref(fig:fig-conditional-effects)b). 
However, suitability is still an important predictor as the WAIC difference between our baseline model and that that did not include suitability was `r round(waic_diff_suitability)` ± `r round(waic_se_diff_suitability)`. 
Both among plants and pollinators for some species there is a strong negative relationship between suitability and normalised degree and some others with a strong positive relationship (Figure \@ref(fig:fig-random-effects)a). 
As expected, we found a positive relationship between a species generalism and normalised degree \@ref(fig:fig-conditional-effects)c). 
However our model comparisons based on WAIC shows that including generalism in the model adds relatively little information once we have accounted for the other variables in the model.
Finally, we found a strong and positive relationship between the number of possible interactions and the number of realised interactions in the community.

```{r fig-conditional-effects, fig.width = width('single'), fig.height = 2.2*3, fig.cap = fig_conditional_effects_legend}
loadd(median_trials)

median_n_pla <- median_trials %>%
  dplyr::filter(guild == "pla_id") %$% 
  n_opposite_guild

median_n_ani <- median_trials %>%
  dplyr::filter(guild == "ani_id") %$% 
  n_opposite_guild

fig_conditional_effects_legend <- 
  paste0("Conditional effects of predictors in our baseline model. The predicted values of the number of interacting species are based on a hypothetical community with ",
         median_n_pla, " plants and ", 
         median_n_ani, " pollinators",
         ". These values correspond to the median number of species in each guild respectively. In each panel we condition on the mean values of all other predictors in the model, mean values for each predictor are indicated with a vertical dashed line. For model fitting, we scaled all predictors to have mean of zero and unit variance. However, with the exception of environmental niche size, here we show the unscaled predictors to facilitate interpretation. To illustrate the uncertainity around the fitted estimates we plot the fits of 100 independent draws from the posterior distribution. The thick lines indicate the mean values of the response distribution. As there was no interaction between guild and generality or the number of possible interactions, for these two predictors we only show conditional the conditional effect of pollinators.")

readd(fig_conditional_effects)
```

```{r tab-model-comparison-table}
model_ranking$waic %>%
  purrr::map(~.$estimates["waic", ]) %>%
  purrr::map_df(~tibble::tibble(estimate = .[1], se = .[2]), .id = "formula") %>%
  dplyr::mutate(formula = translate_model_formula(formula, "long-abv"), 
                formula = kableExtra::cell_spec(formula, format = "latex", 
                                                bold =nchar(formula) == max(nchar(formula)))) %>% 
   knitr::kable("latex", 
               booktabs = TRUE, 
               digits = c(0,0,0), 
               format.args = list(big.mark = ","), 
               col.names = c("predictors", "WAIC", "SE"),
               escape = FALSE,
               linesep = "",
               caption = "Comparison in out of sample predictive power of the baseline model (bold) and their alternatives. We rank models by their expected log predictive density based on their leave-one-out cross-validation information criterion (LOO). The standard error of the LOO difference provides rough guidance to the uncertainty of the model ranking. We also show the Wanatabe-Akaike information criterion (WAIC) of each model for comparison.") %>%
  kableExtra::kable_styling(font_size = 8) #%>%
  # kableExtra::add_header_above(c(" " = 1, "LOO" = 3, " " = 1))
# 
# waic_scores <- as.data.frame(model_ranking$waic) %>%
#   tibble::rownames_to_column(var = "formula") %>%
#   dplyr::select(formula, waic)
# 
# loo_scores <- as.data.frame(model_ranking$loo) %>%
#   tibble::rownames_to_column(var = "formula") %>%
#   dplyr::select(formula, elpd_diff, se_diff,  looic) 
# 
# loo_scores %>%
#   dplyr::inner_join(waic_scores, by = "formula") %>%
#   dplyr::mutate(formula = translate_model_formula(formula, "long-abv"), 
#                 formula = kableExtra::cell_spec(formula, format = "latex", 
#                                                 bold =nchar(formula) == max(nchar(formula)))) %>%
#   knitr::kable("latex", 
#                booktabs = TRUE, 
#                digits = c(0,0,1,0,0), 
#                format.args = list(big.mark = ","), 
#                col.names = c("predictors", "$\\Delta$", "$SE_{\\Delta}$", "IC", "WAIC"),
#                escape = FALSE,
#                linesep = "",
#                caption = "Comparison in out of sample predictive power of the baseline model (bold) and their alternatives. We rank models by their expected log predictive density based on their leave-one-out cross-validation information criterion (LOO). The standard error of the LOO difference provides rough guidance to the uncertainty of the model ranking. We also show the Wanatabe-Akaike information criterion (WAIC) of each model for comparison.") %>%
#   kableExtra::kable_styling(font_size = 8) %>%
#   kableExtra::add_header_above(c(" " = 1, "LOO" = 3, " " = 1))
```

The group level variation among communities was larger than that among species.
The standard deviation (in the parameters scale) of the community intercepts was `r round(sd_loc_id$Estimate, 2)` [`r round(sd_loc_id$Q2.5, 2)`, `r round(sd_loc_id$Q97.5, 2)`] while the standard deviation of the species intercept was `r round(sd_org_id$Estimate, 2)` [`r round(sd_org_id$Q2.5, 2)`, `r round(sd_org_id$Q97.5, 2)`], and that of the species' suitability slope was `r round(sd_org_id_suitability$Estimate, 2)` [`r round(sd_org_id_suitability$Q2.5, 2)`, `r round(sd_org_id_suitability$Q97.5, 2)`] (95% credible intervals shown within square brakets). 
Interestingly, the slope the relationship between suitability and normalised degree was negatively correlated with the species' intercept in the model (Figure \@ref(fig:fig-random-effects)b). 
The mean correlation coefficient was `r round(random_cor$Estimate, 2)` [`r round(random_cor$Q2.5, 2)`, `r round(random_cor$Q97.5, 2)`].

```{r fig-random-effects, fig.width = width('single'), fig.height = 2.2*2.5, fig.cap = fig_random_effects_legend}
loadd(random_species_draws)

n_random_species <- random_species_draws %>%
  dplyr::group_by(guild) %>%
  dplyr::summarise(n_sp = dplyr::n_distinct(org_id))

n_random_pla <- n_random_species %>%
  dplyr::filter(guild == "pla_id")

n_random_ani <- n_random_species %>%
  dplyr::filter(guild == "ani_id")

fig_random_effects_legend <- 
  paste0("Species level effects of suitability. (a) Conditional effect of suitability for individual species. To facilitate visualization we show only species for which there is suitability information in at least six communities (", 
         n_random_pla, " plants and ", 
         n_random_ani, " pollinators). As in the previous figure, fitted values assume a hypothetical community of median size. In each panel we highlight two species for which the relationship between environental suitability and normalised degree was particularly strong. ", 
         "(b) The correlation between the species' intercept and the species' slope of suitability was negatively correlated.")

readd(fig_random_effects)
```

# Discussion 

We found a positive association between environmental suitability and the niche breath. 
This relationship was maintained when we account for the overall generality of a species and even in extremely conservative model parametrization when we include the number of possible interacting species as an independent variable. 
We define "possible" interacting species as one with which the species interacts in any of our ecological communities. 

No interaction significant between guild and relationship which highlights that differences are rooted in eco-evolutionary processes rather than behaviour. 

Relationship is even stronger when we only include species for which we have more info. i.e. in more that 5 communities. 


<!-- We focus suitability in abiotic factors but can also be influenced by dispersal and species interactions themselves (e.g. competition and facilitation, Lortie et al. 2004).  -->
<!-- We learnt that there is a trade-off pollination dynamics (Cagua in review?). -->
<!-- This means that species with more environemntal favours are better prepared to deal with the trade-offs that pollination can convey. Cool! -->

<!-- Suitability is not a meric of abbundances across sites (Pearce & Ferrier 2001; Sagarin et al. 2006) (Taken from Boulangeat) -->

<!-- Remarkable that we found a consistent signal of suitability given our relatively small number of communities and the multitude of processes that shape n. interactions Eltonian niche. We didn;t even account for traits. Interactions can influence co-occurrence (indirect and third order too see refs in Gravel 2018) -->

<!-- Add the suitability of possible competitors? I mean they are not really competitors so maybe not.  -->

<!-- Discuss implications for network rewiring? -->

Note that although abundances can inflence wether two species interact we cannot tease t appart because suitability is not necessarily related to abundance. ALthough abundance affects the frequency of the interaction, rather than wether they interacts or not (Donoso)

# Acknowledgements

`r paste(drake::readd(acknowledgements), collapse = "")`

# References

