---
title: "Environmental stress affects niche breadth in plant-pollinator communities"
author: 
  - "Audrey Lustig^2^ ([audrey.lustig@canterbury.ac.nz](audrey.lustig@canterbury.ac.nz))"
  - "Jason M. Tylianakis^1^ ([jason.tylianakis@canterbury.ac.nz](jason.tylianakis@canterbury.ac.nz))"
  - "Daniel B. Stouffer^1^ ([daniel.stouffer@canterbury.ac.nz](daniel.stouffer@canterbury.ac.nz))"
  - "E. Fernando Cagua^1^ ([efc29@uclive.ac.nz](mailto:efc29@uclive.ac.nz))"
bibliography: 
  - bibliography.bib
  - int-bibliography.bib
# csl: pnas.csl
output: #word_document
  bookdown::pdf_document2:
    # base_format: rticles::peerj_article
    keep_tex: yes
    number_sections: false
toc: false
fontsize: 11pt
# classoption: a4paper
geometry: 
  - textwidth=33em
  - textheight=48\baselineskip
header-includes:
  - \usepackage{booktabs}
  - \usepackage{setspace}
  - \usepackage{lineno}
  - \usepackage{xr}
  - \usepackage[utf8]{inputenc}
  - \newcommand{\R}[1]{\label{#1}\linelabel{#1}}
  - \newcommand{\lr}[1]{page~\pageref{#1}, line~\lineref{#1}}
  - \externaldocument{supp-info}
  - \usepackage[export]{adjustbox}
  - \usepackage{caption}
editor_options: 
  chunk_output_type: console
---

\linenumbers
\doublespacing
\captionsetup[table]{skip=10pt}

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, 
                      warning = FALSE, dpi = 300, 
                      eval.after = "fig.cap")

library(dplyr)

loadd(abs_wordcount, msc_wordcount, n_displays, n_references, keywords, bibliography)
loadd(interaction_citations, config) 
```
 
^1^ Centre for Integrative Ecology, School of Biological Sciences, University of Canterbury, Private Bag 4800, Christchurch 8041, New Zealand

^2^ Geospatial Research Institute, University of Canterbury, Private Bag 4800, Christchurch 8041, New Zealand

**Running title:** XX

**Keywords:** `r keywords`

**Type of article:** XX

**Number of words:** `r abs_wordcount$n_words_stri` in abstract; `r format(round(msc_wordcount$n_words_stri), big.mark = ",")` in main text.

**Number of displays:** `r n_displays$figures` figures; `r n_displays$tables` tables; `r n_displays$text_boxes` text boxes.

**Number of references:** `r n_references + nrow(interaction_citations)`

**Author for correspondence:** E. Fernando Cagua (+64 20 4026 8153).

**Data accessibility:** Data supporting the results will be accessible in an appropriate data repository after publication. We will include the DOI here.

**Author contributions:** XX

\clearpage

# Abstract

`r paste(drake::readd(abstract), collapse = "")`

\clearpage

# Introduction

Species interactions are known to vary widely across space and time. 
There are multiple examples of species that interact with a large number of partners in a particular community or season, but with fewer in another. 

Some of this variation can be attributed to environmental drivers. 
However, how exactly the environment, specifically the stress it imposes on species, affects whether two species interact or not, and ultimately the number of partners a species has is still unknown. 
Understanding how the number of partners—the species degree—is driven by the environment is crucial because it underpins the species role in its community and shapes the structure of the network of interactions. 
This structure, in turn, determines ecosystem function and stability. 

Species interactions are determined in part by niche processes (the matching of traits) and partly by neutral processes (more abundant species are more likely to encounter each other and, thus, interact). 
The environment can influence both of these processes. 
It is, therefore, not surprising that, despite limitations on the spatial extent or the number of environmental gradients considered, multiple studies have been able to show how changes to interactions can be related to environmental change [@tylianakis_ecological_2017].
For instance, some studies suggest that the strength of some trophic interactions, like predation [@mckinnon_lower_2010; @vucic-pestic_warming_2011] and herbivory [@baskett_latitudinal_2018], can increase with temperature but might decrease with precipitation [@pires_predicted_2016]. 
Some other studies, however, have shown either no effect (on average) or non-linear effects of temperature or precipitation on plant-pollinator interactions [@devoto_patterns_2005; @gravel_bringing_2018].
Overall, while it looks clear that pairwise interactions respond to environmental drivers, there is high variability in the response [@tylianakis_global_2008].

One possible explanation for the seemingly contradictory evidence is that each species can have multiple partners. 
Each of these partners, as well as the interactions with them, can be simultaneously affected by the environmental conditions. 
Therefore environmental stress may affect the number of partners in different ways depending on its role in the community (for example its trophic guild) or even the species itself. 
Previous research suggests that there might be two alternative hypotheses of how environmental stress may affect species degree [@tylianakis_ecological_2017]. 
On the one hand, it is possible that when species are under environmental stress, they might be "pressured" to focus on partners with which they are best adapted to interact. 
For instance, @hoiss_altitude_2012 found increased phylogenetic clustering between plants and pollinators at higher altitudes; while @peralta_phylogenetic_2015 found that parasitoids in plantation forest, where environmental stress was higher than in native forests, were constrained to interact with hosts, they were best adapted to attack.
Similarly, @lavandero_genotype_2013 found that environmental stress due to higher temperature reduced the breadth of the Eltonian niche of parasitoids. 

On the other hand, it is also possible that when species are under environmental stress, they are forced to be more flexible in their interactions as higher environmental stress is likely to be reflected in greater energetic or reproductive costs. 
Therefore they might not be able to sustain encounter rates with their preferred partners at sufficient levels. 
In line with this hypothesis, @hoiss_interactive_2015 found that the specialisation of plant-pollinator networks decreased both with elevation and after extreme drought events.
Likewise, @pellissier_spatial_2010 found a positive relationship between niche breadth and environmental stress: disk- or bowl-shaped blossoms (which allow a large number of potential pollinator species to access pollen and nectar rewards) dominated at high altitude flower communities. 

Here, we investigate whether, and how, environmental stress can systematically affect species degree. 
Our main aim is to test the two competing hypotheses that relate environmental stress and species degree and investigate whether this changes across species or between trophic guilds. 
We propose that specialist species can become "facultative" generalists to reduce their vulnerability to the absence of preferred partners [for example, when variations in climate decouple phenologies; @benadi_specialization_2014]. 
We therefore also expect that as environmental stress increases species with a relatively small number of partners are more likely to engage with more partners and broaden their trophic niche. 
Species with a large number of partners, on the other hand, should have a larger pool of available partners and might, therefore, be more likely to narrow their niche under environmental stress by focusing on the most beneficial partners. 
Importantly, when testing these hypotheses, we control for the potential effects of the environment in community composition and the size of the species fundamental niche, both from an Eltonian (interactions) and Grinellian (environment) perspective.

We test these hypotheses using data on plant-pollinator interactions. 
We use the species' patterns of occurrence to estimate the environmental suitability in their communities as an indirect measure of the environmental stress they might experience. 
Condensing the environmental variation over multiple factors (like temperature and precipitation) into a single metric is crucial to generalise our findings at a global scale. 

# Methods

```{r}
loadd(interaction_citations)
int_citations_year_range <- range(interaction_citations$YEAR)

loadd(clean_interactions)
stats_clean_int <- clean_interactions %>%
  purrr::map(n_distinct)
```

We retrieved plant-pollinator networks from the Web of Life database [@fortuna_web_2014]. 
This database contains datasets originating from `r length(interaction_citations$BIBTEXKEY)` studies published in the primary literature between `r int_citations_year_range[1]` and `r int_citations_year_range[2]`. 
Calculating the environmental stress of species in their community and their Eltonian niche breadth required us to reduce both the taxonomic and distributional/locational uncertainty.
A critical step towards reducing this uncertainty is to ensure that the names used to identify species are valid and unambiguous, which in turn allow us to obtain further information from biological databases and accurately match species across studies. 
Therefore, our first step was to ensure consistent spelling and standardisation of species names synonyms (See Supplementary Methods).
This step ensured that the matching of species across studies was as accurate as possible. 
The cleaning process resulted on a total of `r format(stats_clean_int$pla_id, big.mark = ",")` plants and `r format(stats_clean_int$ani_id, big.mark = ",")` pollinator species distributed across `r format(stats_clean_int$loc_id)` locations arround the globe (Figure \@ref(fig:fig-worldmap), \@ref(fig:fig-distribution-species-mult-locations)).

```{r fig-worldmap, fig.width = width('single'), fig.cap ="Worldwide distribution of pollination communities included in this study"}
readd(fig_worldmap)
```

```{r fig-distribution-species-mult-locations, fig.width = width('single'), fig.height = 2.2, fig.cap = fig_dist_species_multiple_locations_legend}

most_popular_species <- readd(fig_dist_species_multiple_locations_data) %>%
  filter(n_locations == max(n_locations)) %>%
  arrange(guild) 

fig_dist_species_multiple_locations_legend <- 
  paste0("Frequency distribution of the number of locations in which a species is present. The most common pollinator species was *", 
       most_popular_species$sp_name[1], 
       "*, which was sampled on ", 
       most_popular_species$n_locations[1], 
       " locations, while the most common plant species was *", 
       most_popular_species$sp_name[2], 
       "*, which was sampled on ", 
       most_popular_species$n_locations[2], 
       " locations.")

readd(fig_dist_species_multiple_locations)
```

Second, we calculated the suitability of the environment for a species in a particular community as a proxy of environmental stress. 
We assume that the environmental stress a species experience in a particular location is inversely related to the suitability of the average environmental conditions in that place. 
Our third and final step was to relate the environmental suitability to the relative number of partners a species has in a community, as a proxy for Eltionian niche breadth. 
To explore this relationship within and across species, we used a multilevel Bayesian model in which we controlled for the potential effects of the environment on co-occurrence.

### Species suitability

```{r}
loadd(n_occurrences, n_cleaned_occurrences, n_dirty_occurrences, gbif_key_groups, org_ids, n_thinned_occurrences, complete_climate_cells, complete_filled_cells_1, complete_filled_cells_2)

get_n_species <- . %>%
  dplyr::inner_join(gbif_key_groups, by = "taxonKey") %>%
  dplyr::inner_join(org_ids, by = c("key_id" = "sp_key_id")) %$%
  dplyr::n_distinct(org_id)

percent_thinned <- 1 - sum(n_thinned_occurrences$N) / sum(n_cleaned_occurrences$N)
percent_filled_round_one <- 1 - complete_filled_cells_1[1]/complete_climate_cells[1]
```

Our next step was to determine the habitat suitability of the species as a proxy of the environmental stress they experience in their community.
As we aim to compare the trophic niche for different suitability levels, we only do this for species that were present in at least `r numbers2words(readd(minimum_spp_locations))` communities.
To calculate the suitability of a species in a particular location, we used a niche-factor analysis [@hirzel_ecological-niche_2002; @broennimann_measuring_2012]. 
This approach is based on the probability density function of species distribution in an environmental variable space. 
In a nutshell, habitats (characterised by a collection of environmental variables) in which the species occurs most often are deemed to be suitable for the species than habitats in which the species has never been observed. 
This approach to estimating the habitat suitability requires two critical pieces of information. 
First, we require information about the occurrences of the species of interest.
Second, we require information about the environmental conditions for all the locations in which the species occurs.

We retrieved `r round(sum(n_occurrences$N)/1000000, digits = 1)` million occurrences from the Global Biodiversity Information Facility (GBIF; https://www.gbif.org). 
Issues with data quality are a central issue hampering the use of publicly available species occurrence GBIF data in ecology and biogeography [@jetz_essential_2019]. 
We, therefore we followed a series of filters and geographic heuristics to correct or remove erroneous and imprecise referencing records [See supplementary methods; @zizka_coordinatecleaner_2019] which allowed us to identify and remove `r round(sum(n_dirty_occurrences$N)/1000000, digits = 1)` million problematic occurrences from further analysis.
We integrated the occurrences from our plant-pollinator communities to the cleaned occurrences retrieved from GBIF.

We retrieved environmental data from WorldClim V2.0, which includes 19 bioclimatic variables commonly used in species distribution modelling [@fick_worldclim_2017]. 
We then complemented data obtained from WorldClim with data from Envirem [@title_envirem_2017], which includes 16 extra bioclimatic and two topographic variables. 
The additional set of variables from Envirem are relevant to ecological or physiological processes and as such, have the potential to improve our suitability estimation [@title_envirem_2018]. 
We obtained all environmental data as rasters composed by cells of 2.5 arc-minutes. 
We chose this resolution because it provides a reasonable match to the locational accuracy of the species occurrences found in GBIF, particularly those that come preserved specimens in museum collections. 

After obtaining information about species occurrence and the environment, we then merged these two datasets such that a vector with details of our 37 bioclimatic and topographic variables characterised the location of each occurrence. 

Sets of occurrence data tend to be spatially aggregated due to sample bias (tendency to collect close to cities, certain countries). 
Spatial autocorrelation arises in ecological data because geographically clumped records tend to be more similar, in physical characteristics and/or species abundances, than are pairs of locations that are farther apart. 
To account for such spatial dependency in occurrence data, if a species had more than one occurrence records within one of the cells of the bioclimatic raster, we only included one of the occurrence records. 
We did this to avoid giving more weight to areas with a high number of occurrences, a common scenario in occurrence records collected opportunistically as the ones we use here. 
In this step we removed `r scales::percent(percent_thinned)` of the occurrences which resulted in a total of `r round(sum(n_thinned_occurrences$N)/1000000, digits = 1)` million occurrences used in our niche analysis. 

A common issue of terrestrial bioclimatic datasets is that the boundaries of the cells with information do not precisely match the landmass boundaries. 
The result of this missmatch is that not all environmental variables was not available for `r format(complete_climate_cells[1], big.mark = ",")` of the raster cells with occurrences (`r scales::percent(complete_climate_cells[1]/sum(complete_climate_cells), accuracy = 0.1)` of the total). 
As expected, the vast majority of these problematic cells were close to the shore. 
To address this issue, we calculated the average value of environmental variables within an `r config$climate_buffer/1000`km buffer of the centre of the cell where the variable was missing and used it to approximate the value of the variable in that cell. 
Using this procedure, we were able to fill environmental variables for `r scales::percent(percent_filled_round_one, accuracy = 0.1)` of the cells were they were missing. 
To fill the remaining `r format(complete_filled_cells_1[1], big.mark= ",")` cells, we repeated the aforementioned procedure but instead using a `r config$climate_buffer*2/1000`km buffer. 
We removed from further analysis occurrences located within the `r complete_filled_cells_2[1]` cells for which we were unable to fill environmental variables. 

Next, we calculate the probability density function of the species distribution in environmental space. 
To determine the environmental space, we use the first two components from a principal component analysis of the 37 bioclimatic variables associated with the species occurrences. 
Specifically we use the `dudi.pca` function from the R package `ade4` `r as.character(packageVersion('ade4'))` [@dray_ade4_2007] and center and scale all bioclimatic variables to have a mean of 0 and a unit standard deviation. 
We then determine the position of species occurrences in the environmental space and estimate their bivariate probability density function. 
We use a kernel method to estimate this density and normalise it such that it ranges between zero and one. 
Specifically, to calculate the probability density function we use `ecospat.grid.clim.dyn` from the R package `ecospat` `r as.character(packageVersion('ecospat'))` [@broennimann_ecospat_2018] with a grid resolution of 200.
We then determine the location in the environmental space of the plant-pollinator communities using the function `suprow` from `ade4`.
The normalised density at that particular location corresponds to our suitability metric, which we calculate using the R package `raster` `r as.character(packageVersion('raster'))` [@hijmans_raster_2019]. 
We use the kernel density method in the niche-factor analysis [@broennimann_measuring_2012] rather than the distance from the mode [@hirzel_ecological-niche_2002], as it has been proposed earlier, as it has been shown to reduce the procedure's sensitivity to sampling effort and the resolution of the environmental space.

```{r sensitivity-analysis-calculations}
loadd(sensitivity_species_name)
loadd(n_enough_occurrences)
n_occ_sensitivity_species <- n_thinned_occurrences %>%
  dplyr::filter(org_id == sensitivity_species_name$org_id[1]) %$% N

n_spp_not_enough_occ <- nrow(n_thinned_occurrences) - nrow(n_enough_occurrences)
```

We used a sensitivity analysis to determine the minimum number of occurrences that are necessary to have robust environmental suitability values in our communities. 
For that we used the species with most occurrences available, *`r sensitivity_species_name$canonicalName`*, and calculated the mean absolute error of the suitability values obtained with `r numbers2words(config$n_subsamples)` subsamples from the `r format(n_occ_sensitivity_species, big.mark  =",")`  occurrences available from GBIF. 

## Data analysis

We then used a set of bayesian multilevel models to evaluate the impact of environmental suitability on the number of partners a species has. 
Specifically, we use the normalised degree of species as our response variable; this is, the number of species it interacts with given the number of species in the opposite guild [@martin_gonzalez_centrality_2010]. 
The normalised degree was modelled using a logit link function, and a binomial distribution in which the number of species interacts with is the number of successes, and the number of species in the opposite guild is the number of trials. 
We are aware that whether species interact or not is not a Bernoulli process as species interactions are not strictly independent from each other. 
However, a binomial distribution allows us to account for the differences in species richness across communities indirectly. 
Importantly, however, results are qualitatively similar when we model species degree directly using a Poisson distribution and a logarithmic link function.

We evaluate four models to assess the relative importance of suitability. 
A first model, our baseline model, included three population-level predictors and two grouping levels, species and the community.
The population-level predictors in the baseline model, commonly called fixed effects, were the habitat suitability, the species guild (plant or a pollinator), and its number of known possible partners.
We included the number of known possible partners as a predictor in our models as it allows us to control for the environmental effects on species co-occurrence. 
We calculate this metric by determining the number of partners with which the species is known to interact in any other community.
Controlling for the number of potential partners makes our model a particularly stringent test of our environmental stress hypotheses because this variable could explain a large proportion of variance. 
Often the potential and the actual number of partners is the same or very close to each other, especially for rare species present only in a few communities.
As we were interested in understanding whether the effect of habitat suitability is conditional on the species guild (plant or pollinator), we, therefore, included guild and its interaction with suitability in the model. 
We allowed the intercept of degree and slope of the suitability-degree relationship to vary among species. 
This approach allowed us to investigate two questions. 
First, it allows us to inspect the extent to which suitability is a population or a group level effect. 
Second, by investigating the correlation between the intercept and the slope as a model parameter, it allowed us to inspect the extent by which species with a small or large number of interactions respond to increasing levels of environmental stress. 
To account for unmeasured differences between communities, like sampling effort, sampling method, or diversity, we also calculated an intercept for each community in our study.
To facilitate model interpretation and convergence, we scaled all continuous variables to have a mean of zero and a unit standard deviation. 

We compared this baseline model with three alternative models in which we remove one predictor at a time.
To quantify the difference between models, in terms of their expected out-of-sample performance, we use the Wanatabe-Akaike information criterion (WAIC).
All models were fitted under a bayesian framework using the R package `brms` `r as.character(packageVersion('brms'))` [@burkner_brms_2017; @burkner_advanced_2018] as an interface for Stan [@carpenter_stan_2017]. 
For each model, we used four Markov chains of 4,000 iterations each; we used half of the iterations for warmup.
We used weakly informative priors for all model parameters. 
Specifically we used normal priors of mean zero and standard deviation ten for the population-level effects and the intercepts, a half-Cauchy prior with a location of zero and a scale of two for the standard deviations, and, when applicable, an LKJ-correlation prior with parameter \(\zeta = 1\) for the correlation matrix between group-level parameters.

# Results

After performing our sensitivity analysis, we found that we need roughly `r readd(min_occurrences_factor)` independent occurrences for each community for which we calculated a suitability value in order to obtain a mean absolute error below `r config$min_suitability_error` (Fig. \@ref(fig:fig-sensitivity-analysis)). 
We, therefore, removed from further analyses `r n_spp_not_enough_occ` species for which we did not have enough occurrences to obtain robust estimates.
When inspecting the suitability values of the analysed species, we found that most communities included species for which habitat suitability was low and species for which it was high (Fig. \@ref(fig:fig-median-suitability)).

```{r model-results-calculations}
loadd(bayesian_r2_baseline)
loadd(parameter_posterior_summary)
loadd(model_ranking)

random_cor <- parameter_posterior_summary %>%
  dplyr::filter(parameter == "cor_org_id__Intercept__scaled_suitability")

sd_loc_id <- parameter_posterior_summary %>% 
  dplyr::filter(stringr::str_detect(parameter, "sd_loc_id__Intercept"))

sd_org_id <- parameter_posterior_summary %>% 
  dplyr::filter(stringr::str_detect(parameter, "sd_org_id__Intercept"))

sd_org_id_suitability <- parameter_posterior_summary %>% 
  dplyr::filter(stringr::str_detect(parameter, "sd_org_id__scaled_suitability"))

waic_comp_suitability <- model_ranking$waic$formula_no_suitability$pointwise - model_ranking$waic$formula_base$pointwise 

waic_diff_suitability <- sum(waic_comp_suitability[, "waic"])

waic_se_diff_suitability <- sqrt(nrow(waic_comp_suitability) * var(waic_comp_suitability[, "waic"]))
```

Our models performed relatively well.
The bayesian R-squared for our baseline model was `r round(bayesian_r2_baseline[1], digits = 2)`, which indicates our models were able to capture a large proportion of the variability on the data. 
Overall we found that environmental suitability does not show a consistent pattern across species. 
Indeed, when looking at the population level effects, suitability has virtually no relationship with the normalised degree, neither for plants or pollinators (Figure \@ref(fig:fig-conditional-effects)b). 
However, suitability is an important predictor as the WAIC difference between our baseline model and that that did not include suitability was `r round(waic_diff_suitability)` ± `r round(waic_se_diff_suitability)` (Table \@ref(tab:tab-model-comparison)).
This apparent discrepancy can be explained by the variability of the suitability-degree relationship across species.

```{r fig-conditional-effects, fig.width = width('single'), fig.height = 2.2*3, fig.cap = fig_conditional_effects_legend}
loadd(median_trials)

median_n_pla <- median_trials %>%
  dplyr::filter(guild == "pla_id") %$% 
  n_opposite_guild

median_n_ani <- median_trials %>%
  dplyr::filter(guild == "ani_id") %$% 
  n_opposite_guild

fig_conditional_effects_legend <- 
  paste0("Conditional effects of predictors in our baseline model. The predicted values of the number of interacting species are based on a hypothetical community with ",
         median_n_pla, " plants and ", 
         median_n_ani, " pollinators",
         ". These values correspond to the median number of species in each guild, respectively. In each panel, we condition on the mean values of all other predictors in the model. We indicate mean values for each predictor with a vertical dashed line. For model fitting, we scaled all predictors to have a mean of zero and unit variance. However, except for environmental niche size, here we show the unscaled predictors to facilitate interpretation. To illustrate the uncertainty around the fitted estimates, we plot the fits of 100 independent draws from the posterior distribution. The thick lines indicate the mean values of the response distribution. As there was no interaction between guild and generality or the number of possible interactions, for these two predictors, we only show conditional the conditional effect of pollinators.")

readd(fig_conditional_effects)
```

```{r tab-model-comparison}
model_ranking$waic %>%
  purrr::map(~.$estimates["waic", ]) %>%
  purrr::map_df(~tibble::tibble(estimate = .[1], se = .[2]), .id = "formula") %>%
  dplyr::mutate(formula = translate_model_formula(formula, "long-abv"), 
                formula = kableExtra::cell_spec(formula, format = "latex", 
                                                bold =nchar(formula) == max(nchar(formula)))) %>% 
   knitr::kable("latex", 
               booktabs = TRUE, 
               digits = c(0,0,0), 
               format.args = list(big.mark = ","), 
               col.names = c("predictors", "WAIC", "SE"),
               escape = FALSE,
               linesep = "",
               caption = "Comparison in out of sample predictive power of the baseline model (bold) and their alternatives. We rank models by their expected log predictive density based on their leave-one-out cross-validation information criterion (LOO). The standard error of the LOO difference provides rough guidance to the uncertainty of the model ranking. We also show the Wanatabe-Akaike information criterion (WAIC) of each model for comparison.") %>%
  kableExtra::kable_styling(font_size = 8) #%>%
```

While for some species there is a strong negative relationship between suitability and normalised degree, for some others with a strong positive relationship (Figure \@ref(fig:fig-random-effects)a). 
Interestingly, the slope of this relationship correlates negatively with the species' intercept in the model (Figure \@ref(fig:fig-random-effects)b). 
The mean correlation coefficient was `r round(random_cor$Estimate, 2)` [`r round(random_cor$Q2.5, 2)`, `r round(random_cor$Q97.5, 2)`].
In other words, the slope of the suitability-degree relationship was more likely to be positive for species with a smaller number of partners in lower-suitability and negative for species with a larger number of partners in lower-suitability conditions.

```{r fig-random-effects, fig.width = width('double'), fig.height = 2.2*2, fig.cap = fig_random_effects_legend, out.width = "6.81in", out.height = "4.4in", out.extra="center"}
loadd(random_species_draws)

n_random_species <- random_species_draws %>%
  dplyr::group_by(guild) %>%
  dplyr::summarise(n_sp = dplyr::n_distinct(org_id))

n_random_pla <- n_random_species %>%
  dplyr::filter(guild == "pla_id") %$%
  n_sp

n_random_ani <- n_random_species %>%
  dplyr::filter(guild == "ani_id") %$%
  n_sp

fig_random_effects_legend <- 
  paste0("Species-level effects of suitability. (a) Conditional effect of suitability for individual species. To facilitate visualization we show only species for which there is suitability information in at least six communities (", 
         n_random_pla, " plants and ", 
         n_random_ani, " pollinators). As in the previous figure, fitted values assume a hypothetical community of median size. In each panel, we highlight two species for which the relationship between environmental suitability and the normalised degree was particularly strong. ", 
         "(b) The correlation between the species' intercept and the species' slope of suitability was negatively correlated.")

readd(fig_random_effects)
```

As expected, we found a strong and positive relationship between the number of possible interactions and the number of realised interactions in the community. 
There was a large difference on WAIC between the model that included this predictor and that that excluded it. 
This result indicates that the availability of potential partners accounts for a large proportion of the variability on species degree.  
Importantly, our findings of the suitability-degree relationship were qualitatively unchanged whether we included this variable or not. 

The group-level variation among communities was larger than that among species which further indicates the importance of the local context when determining species degree.
Specifically, the standard deviation (in the parameters scale) of the community intercepts was `r round(sd_loc_id$Estimate, 2)` [`r round(sd_loc_id$Q2.5, 2)`, `r round(sd_loc_id$Q97.5, 2)`] while the standard deviation of the species intercept was `r round(sd_org_id$Estimate, 2)` [`r round(sd_org_id$Q2.5, 2)`, `r round(sd_org_id$Q97.5, 2)`], and that of the species' suitability slope was `r round(sd_org_id_suitability$Estimate, 2)` [`r round(sd_org_id_suitability$Q2.5, 2)`, `r round(sd_org_id_suitability$Q97.5, 2)`] (95% credible intervals shown within square brakets). 

# Discussion 

We set out to explore whether and how environmental stress can systematically affect species degree. 
After accounting for the pool of potential partners, we found that indeed environmental conditions contribute to determining whether a species is a generalist or a specialist in their community. 
However, we also found that the particular effect of the environment is strongly dependent on the species.
We proposed two alternative hypotheses of how environmental stress may affect the degree, and we found evidence for both of them.
Species with a large number of partners in low-stress communities were more likely to have a negative relationship and hence reduce the number of partners as stress increases. 
Contrastingly, species in our datasets with a small number of partners in low-stress communities were more likely to have a larger number of partners in more stressful communities.
In summary, environmental stress pushes species that are flexible enough to change their interaction partners towards intermediate levels of degree, a so-called "regression towards the mean".

Our results suggest that changes in community composition are indeed the primary channel through which the environment determines changes interaction probability. 
However, they also show that, for a large number of species, the environment may also play a substantial role in determining their realised (Eltonian) niche. 
While previous research has recognised that environmental factors may help explain the changes in network structure along environmental gradients that cannot be explained by community composition [@tylianakis_habitat_2007], how these two factors were linked had been elusive so far [@gravel_bringing_2018]. 
We believe that part of this difficulty could have arisen because species, and ultimately network structure, can respond in multiple and contrasting, ways depending on the particular bioclimatic variable examined (e.g. temperature or precipitation).
Using stress to summarise the effect on species of multiple environmental gradients allowed us to detect a clear signal of the environment in species' interaction patterns.

Although both niche and neutral processes are relevant at determining species interactions, our model suggests that niche processes may be the predominant mechanism through which the environment *systematically* affect species degree. 
First, it is unlikely that environmental suitability correlates to local species abundances [@pearce_practical_2001; @sagarin_moving_2006]. 
Second, even if there is a relationship between suitability and abundances, a particular environmental gradient could have a positive effect on the abundance of some species and a negative effect on others. 
Indeed, we show that within a community there is a wide range of suitability values, even for the relatively limited number of species we were able to include in our analysis. 

Recent research suggests that species are continuously changing their interaction partners wherever environmental conditions change in space or time [@raimundo_adaptive_2018].
So far it appears that this rewiring is primarily driven by generalist species [@ponisio_opportunistic_2017; @burkle_plant-pollinator_2013], presumably because generalist species are less sensitive to trait matching of their interaction partners [@caradonna_interaction_2017].
Our results add two important nuances to these findings. 
First, because generalists seem to focus on a smaller number of partners as environmental conditions deteriorate, we show that trait matching might still play a role in determining the interactions of generalist species. 
Second, we demonstrate that rewiring is not exclusive to generalists.
At least a fraction of the species that appear to be specialist in their communities might be as flexible, if not more, than generalist species, effectively behaving as facultative generalists in the face of environmental change. 
These flexible specialists might, therefore, have a more significant role in network persistence than previously expected.

While our model detected how the environment affects the interactions of two types of species, generalists and flexible specialists, there is a third group that remained invisible for our model but has important implications for network persistence and stability.
Species that are able to vary their interaction partners flexibly and their role in the network are more likely to persist in their community as environmental conditions vary [@gaiarsa_interaction_2019].
We propose this third group of is composed of specialists that are constrained to interact with partners of high trait-matching and therefore were not likely to be found in more than one community. 
If species that are not flexible are unlikely to persist over temporal or spatial environmental gradients, we can expect specialised communities that are highly constrained by trait-matching [like some plant-hummingbird networks; @vizentin-bugoni_processes_2014; @maruyama_morphological_2014] to be far more vulnerable to increased climate change-induced environmental stress and habitat degradation than communities where role and interaction flexibility are more prevalent.

Similarly, if the patterns we see in our models have also played a role during the evolutionary history of pollination communities, our results also help explain why only a small fraction of plant-pollinator interactions show a strong signature of deep co-evolutionary history [@hutchinson_cophylogenetic_2017].
The increases on the stress that species are predicted to experience due to rapid environmental change might further erode the co-evolutionary history of specialist species.
Communities as a whole might be in a trajectory of even more diffuse co-evolution.
For specialist species, at least, the longer-term benefits of being able to interact with multiple partners might be more important than the shorter-term benefits of interacting with partners of high trait matching. 

The structural implications of the "regression towards the mean" that environmental stress promotes are less clear. 
However, it is plausible to expect that nestedness, and therefore network stability, might be reduced in the face of rapid environmental change.
Determining exactly how the changes in degree caused by environmental stress reflect on systematic changes in network structure would be an interesting avenue of research. 
Answering this question would require expanding our suitability analysis to all species in the community and compare the degree distribution of networks along a gradient of stress for the community as a whole. 

In conclusion, we show that the environment can affect the realised niche of plants and pollinators in systematic ways beyond community composition. 
On the one hand, species that are inflexible with their interaction partners are unlikely to persist under more stressful environmental conditions. 
On the other, species that are flexible with their interactions partners experience a regression towards intermediate levels of degree, where generalist species tend to interact with fewer, presumably more effective, partners and specialist tend to interact with more partners. 

# Acknowledgements

`r paste(drake::readd(acknowledgements), collapse = "")`

# References

