---
title: "Environmental stress drives intermediate plant and pollinator specialisation"
author: 
  - "Audrey Lustig^2^ ([audrey.lustig@canterbury.ac.nz](audrey.lustig@canterbury.ac.nz))"
  - "Jason M. Tylianakis^1^ ([jason.tylianakis@canterbury.ac.nz](jason.tylianakis@canterbury.ac.nz))"
  - "Daniel B. Stouffer^1^ ([daniel.stouffer@canterbury.ac.nz](daniel.stouffer@canterbury.ac.nz))"
  - "E. Fernando Cagua^1^ ([efc29@uclive.ac.nz](mailto:efc29@uclive.ac.nz))"
bibliography: 
  - bibliography.bib
  - int-bibliography.bib
csl: ecology-letters.csl
output: #word_document
  bookdown::pdf_document2:
    # base_format: rticles::peerj_article
    keep_tex: yes
    number_sections: false
toc: false
fontsize: 11pt
# classoption: a4paper
# geometry: 
  # - textwidth=33em
  # - textheight=48\baselineskip
header-includes:
  - \usepackage{booktabs}
  - \usepackage{setspace}
  - \usepackage{lineno}
  - \usepackage{xr}
  - \usepackage[utf8]{inputenc}
  - \newcommand{\R}[1]{\label{#1}\linelabel{#1}}
  - \newcommand{\lr}[1]{page~\pageref{#1}, line~\lineref{#1}}
  - \externaldocument{supp-info}
  - \usepackage[export]{adjustbox}
  - \usepackage{caption}
editor_options: 
  chunk_output_type: console
---

\linenumbers
\doublespacing
\captionsetup[table]{skip=10pt}

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, 
                      warning = FALSE, dpi = 300, 
                      eval.after = "fig.cap")

library(dplyr)

loadd(abs_wordcount, msc_wordcount, n_displays, n_references, keywords, bibliography)
loadd(interaction_citations, config) 
loadd(bibliography)
```
 
^1^ Centre for Integrative Ecology, School of Biological Sciences, University of Canterbury, Private Bag 4800, Christchurch 8041, New Zealand

^2^ Geospatial Research Institute, University of Canterbury, Private Bag 4800, Christchurch 8041, New Zealand

**Running title:** Environmental stress drives specialisation.

**Keywords:** `r keywords`

**Type of article:** Letter.

**Number of words:** `r abs_wordcount$n_words_stri` in abstract; `r format(round(msc_wordcount$n_words_stri), big.mark = ",")` in main text.

**Number of displays:** `r n_displays$figures` figures; `r n_displays$tables` tables; `r n_displays$text_boxes` text boxes.

**Number of references:** `r n_references + nrow(interaction_citations)`

**Author for correspondence:** E. Fernando Cagua (+64 20 4026 8153).

**Data accessibility:** Data supporting the results will be accessible in an appropriate data repository after publication. We will include the DOI here.

**Author contributions:** EFC conceived the study. EFC and AL compiled cleaned the data. EFC analysed the data. All authors contributed to the writing of the manuscript and study design.

\clearpage

# Abstract

`r paste(drake::readd(abstract), collapse = "")`

\clearpage

# Introduction

Species interactions are known to vary widely across space and time [@laliberte_cascading_2012; @trojelsgaard_geographical_2015; @schleuning_specialization_2012; @poisot_beyond_2015]. 
There are multiple examples of species that interact with a large number of partners in a particular community or season, but with fewer in another [@olesen_temporal_2008; @dupont_spatio-temporal_2009; @olesen_strong_2011; @rabeling_seasonal_2019; @benadi_specialization_2014]. 
Some of this variation can be attributed to environmental drivers [@tylianakis_ecological_2017]. 
However, it remains unknown how generalisable characteristics of the environment, specifically the stress it imposes on species, affects whether two species interact, and, ultimately, the species' specialisation. 
Understanding how the environment drives the number of partners is crucial because it underpins the species' role in its community and shapes the structure of the network of interactions [@cirtwill_review_2018]. 
This structure, in turn, determines ecosystem function and stability [@thebault_stability_2010]. 

Species interactions are determined in part by niche processes (the matching of traits) and partly by neutral processes (more abundant species are more likely to encounter each other and, thus, interact) [@vazquez_evaluating_2009; @vazquez_uniting_2009]. 
The environment can influence both of these processes [@godsoe_integrating_2017]. 
It is, therefore, not surprising that, despite limitations on the spatial extent or the number of environmental gradients considered, multiple studies have shown how changes to interactions can be related to environmental change [@tylianakis_ecological_2017].
Despite widespread evidence that species interactions and their networks can change along environmental gradients, the direction of change can be variable [@pires_predicted_2016; @baskett_latitudinal_2018; @devoto_patterns_2005], such that until now it has been impossible to extrapolate previous findings to new kinds of gradients.
Overall, while it looks clear that pairwise interactions respond to environmental drivers, there is high variability in the response [@tylianakis_global_2008].

One possible explanation for the seemingly contradictory evidence is that different bioclimatic factors (like temperature or precipitation) can have contrasting effects on species and their partners, such that we lack a common currency with which to derive general responses to the environment. 
Here we attempt to overcome these barriers to deriving general rules by reducing multiple factors into a single measure of environmental stress. 
Previous research suggests that environmental stress may affect the number of interaction partners a species has in different ways depending on the species' role in the community (for example its trophic guild) [@tylianakis_ecological_2017]. 
Specifically, we propose two alternative hypotheses of how environmental stress may affect specialisation. 
First, species under environmental stress might be "pressured" to focus on partners with which they are best adapted to interact. 
For instance, @hoiss_altitude_2012 found increased phylogenetic clustering between plants and pollinators at higher altitudes; while @peralta_phylogenetic_2015 found that parasitoids in plantation forest, where environmental stress was higher than in native forests, constrained their host use to those with which phylogenetic matching was strongest.
Similarly, @lavandero_genotype_2013 showed that environmental stress due to higher temperature reduced the trophic niche breadth (range of host genotypes used) of parasitoids, suggesting higher specialisation. 

Alternatively, it is also possible that when species are under environmental stress, they are forced to be more flexible in their interactions.
Higher environmental stress is likely to be reflected in greater energetic or reproductive costs. 
Therefore, they might not be able to sustain encounter rates with their preferred partners at sufficient levels. 
In line with this hypothesis, @hoiss_interactive_2015 found that the specialisation of plant-pollinator networks decreased both with elevation and after extreme drought events.
Likewise, @pellissier_spatial_2010 found a positive relationship between niche breadth and environmental stress: disk- or bowl-shaped blossoms (which allow a large number of potential pollinator species to access pollen and nectar rewards) dominated at high altitude flower communities. 

Here, we investigate whether and how environmental stress can systematically affect specialisation. 
Our main aim is to test the two hypotheses mentioned above that relate environmental stress with a species' number of partners and investigate whether this changes systematically across species or between trophic guilds. 
We propose that specialist species can become "facultative" generalists to reduce their vulnerability to the absence of preferred partners [for example, when variations in climate decouple phenologies; @benadi_specialization_2014]. 
In other words, we expect that, as environmental stress increases, specialists should be more likely to engage with more partners. 
Species with many partners, conversely, should have a larger pool of available partners and might, therefore, be more likely to specialise on the most beneficial partners under environmental stress. 
Importantly, when testing these hypotheses, we control for the potential effects of the environment on community composition [which has been previously shown to be a determinant factor; @gravel_bringing_2018]. 
We test these hypotheses using data on plant-pollinator interactions. 
These interactions provide a particularly interesting system to test these hypotheses because, due to the multiple trade-offs involved in the pollination service, there are multiple intuitive ways in which we could imagine species responding to environmental stress given the available partners.
We estimate the stress species might experience a given community by calculating its bioclimatic suitability based on the species' patterns of global occurrence. 

# Methods

```{r}
loadd(interaction_citations)
int_citations_year_range <- range(interaction_citations$YEAR)

loadd(clean_interactions)
stats_clean_int <- clean_interactions %>%
  purrr::map(n_distinct)
```

We retrieved plant-pollinator networks from the Web of Life database [@fortuna_web_2014]. 
This database contains datasets originating from `r length(interaction_citations$BIBTEXKEY)` studies published in the primary literature between `r int_citations_year_range[1]` and `r int_citations_year_range[2]`. 
Calculating the environmental stress of species in their community and their potential partners required us to reduce both the taxonomic and distributional/locational uncertainty.
A critical step towards reducing this uncertainty is to ensure that the names used to identify species are valid and unambiguous, which in turn allows us to obtain further information from biological databases and accurately match species across studies. 
Therefore, our first step was to ensure consistent spelling and standardisation of species names and synonyms (see Supplementary Methods).
The cleaning process resulted on a total of `r format(stats_clean_int$pla_id, big.mark = ",")` plants and `r format(stats_clean_int$ani_id, big.mark = ",")` pollinator species distributed across `r format(stats_clean_int$loc_id)` locations arround the globe (Figure \@ref(fig:fig-distribution-species-mult-locations) and  \@ref(fig:fig-worldmap)).

```{r fig-distribution-species-mult-locations, fig.width = width('single'), fig.height = 2.2, fig.cap = fig_dist_species_multiple_locations_legend}

most_popular_species <- readd(fig_dist_species_multiple_locations_data) %>%
  filter(n_locations == max(n_locations)) %>%
  arrange(guild) 

fig_dist_species_multiple_locations_legend <- 
  paste0("Frequency distribution of the number of locations in which a species is present. The most common pollinator species was *", 
       most_popular_species$sp_name[1], 
       "*, which was sampled at ", 
       most_popular_species$n_locations[1], 
       " locations, while the most common plant species was *", 
       most_popular_species$sp_name[2], 
       "*, which was sampled at ", 
       most_popular_species$n_locations[2], 
       " locations.")

readd(fig_dist_species_multiple_locations)
```

After matching species across studies as accurately as possible, we performed two more steps. 
First, we calculated the environmental stress of species in their communities.
Second, we relate each species' stress in its community with the number of partner species it has as a metric of its level of specialisation.

## Environmental stressdispla

```{r occurrences-calcs}
loadd(n_occurrences, n_cleaned_occurrences, n_dirty_occurrences, gbif_key_groups, org_ids, n_thinned_occurrences, complete_climate_cells, complete_filled_cells_1, complete_filled_cells_2)

get_n_species <- . %>%
  dplyr::inner_join(gbif_key_groups, by = "taxonKey") %>%
  dplyr::inner_join(org_ids, by = c("key_id" = "sp_key_id")) %$%
  dplyr::n_distinct(org_id)

percent_thinned <- 1 - sum(n_thinned_occurrences$N) / sum(n_cleaned_occurrences$N)
percent_filled_round_one <- 1 - complete_filled_cells_1[1]/complete_climate_cells[1]
```

We calculated the environmental stress of species in their communities.
We assume that the stress a species experiences in a particular location is inversely related to the suitability of the average environmental conditions in that place. 
As we aim to compare specialisation levels for different levels of environmental stress, we only calculate bioclimatic suitability for species that were present in at least `r numbers2words(readd(minimum_spp_locations))` communities.
To calculate the bioclimatic suitability of a species in a particular location, we used a niche-factor analysis [@hirzel_ecological-niche_2002; @broennimann_measuring_2012]. 
This approach is based on the probability density function of species distribution in an environmental variable space. 
Habitats are characterised by a collection of environmental variables. 
In a nutshell, those habitats in which the species occurs more often are deemed to be more suitable for the species than habitats in which the species has never been observed. 
As bioclimatic suitability is calculated in a scale from zero to one following the niche-factor analysis, for simplicity, we define environmental stress as one minus suitability. 

The niche factor analysis requires two critical pieces of information. 
First, it requires information about the occurrences of the species of interest.
Second, the method requires information about the environmental conditions for all the locations in which the species occurs.
We retrieved `r round(sum(n_occurrences$N)/1000000, digits = 1)` million occurrences of our focal species from the Global Biodiversity Information Facility (GBIF; https://www.gbif.org). 
Issues with data quality are a central issue hampering the use of publicly available species occurrence GBIF data in ecology and biogeography [@jetz_essential_2019]. 
We, therefore, followed a series of filters and geographic heuristics to correct or remove erroneous and imprecise referencing records [see supplementary methods; @zizka_coordinatecleaner_2019] which allowed us to identify and remove `r round(sum(n_dirty_occurrences$N)/1000000, digits = 1)` million potentially problematic occurrences from further analysis.
We integrated the occurrence records from our plant-pollinator communities to the cleaned occurrences retrieved from GBIF.

We retrieved environmental data from WorldClim V2.0, which includes 19 bioclimatic variables commonly used in species distribution modelling [@fick_worldclim_2017]. 
We then complemented data obtained from WorldClim with data from Envirem [@title_envirem_2017], which includes 16 extra bioclimatic and two topographic variables. 
The additional set of variables from Envirem are relevant to ecological or physiological processes and thus have the potential to improve our suitability estimation [@title_envirem_2018]. 
We obtained all environmental data as rasters composed by cells of 2.5 arc-minutes. 
We chose this resolution because it provides a reasonable match to the locational accuracy of the species occurrences found in GBIF, particularly those that originate from preserved specimens in museum collections. 

After obtaining information about species occurrence and the environment, we then merged these two datasets such that a vector with details of our 37 bioclimatic and topographic variables characterised the location of each occurrence. 
Sets of occurrence data tend to be spatially aggregated due to sample bias (tendency to collect close to cities, certain countries). 
Moreover, spatial autocorrelation arises in ecological data because geographically clumped records tend to be more similar in physical characteristics and/or species abundances than do pairs of locations that are farther apart. 
To account for such spatial dependency in occurrence data, we only included one occurrence record if a species had more than one within a cell of the bioclimatic raster. 
We did this to avoid giving more weight to areas with a high number of occurrences, a common scenario in occurrence records collected opportunistically as the ones we use here. 
In this step we removed `r scales::percent(percent_thinned)` of the occurrences which resulted in a total of `r round(sum(n_thinned_occurrences$N)/1000000, digits = 1)` million occurrences used in our niche analysis. 

A common issue of terrestrial bioclimatic datasets is that the boundaries of the cells with information do not precisely match the landmass boundaries. 
The result of this missmatch is that not all environmental variables were available for `r format(complete_climate_cells[1], big.mark = ",")` of the raster cells with occurrences (`r scales::percent(complete_climate_cells[1]/sum(complete_climate_cells), accuracy = 0.1)` of the total). 
As expected, the vast majority of these problematic cells were close to the shore. 
To address this issue, we calculated the average value of environmental variables within a `r config$climate_buffer/1000`km buffer of the centre of the cell where the variable was missing and used it to approximate the value of the variable in that cell. 
Using this procedure, we were able to fill environmental variables for `r scales::percent(percent_filled_round_one, accuracy = 0.1)` of the cells where they were missing. 
To fill the remaining `r format(complete_filled_cells_1[1], big.mark= ",")` cells, we repeated the aforementioned procedure but instead using a `r config$climate_buffer*2/1000`km buffer. 
We removed from further analysis occurrences located within the `r complete_filled_cells_2[1]` cells for which we were unable to fill environmental variables (`r scales::percent(complete_filled_cells_2[1]/sum(complete_filled_cells_2), 0.01)` of the total). 

Next, we calculated the probability density function of the species distribution in environmental space. 
To determine the environmental space, we used the first two components from a principal component analysis of the 37 bioclimatic variables associated with the species occurrences. 
Specifically we used the `dudi.pca` function from the R package `ade4` `r as.character(packageVersion('ade4'))` [@dray_ade4_2007] and centered and scaled all bioclimatic variables to have a mean of zero and a unit variance. 
We then determined the position of species occurrences in the environmental space and estimated their bivariate probability density function. 
We used a kernel method to estimate this density and normalised it such that it ranges between zero and one. 
We used the kernel density method in the niche-factor analysis [@broennimann_measuring_2012] rather than the distance from the mode [@hirzel_ecological-niche_2002] (as has been proposed earlier) because it has been shown to reduce the procedure's sensitivity to sampling effort and the resolution of the environmental space.
Specifically, to calculate the probability density function we used `ecospat.grid.clim.dyn` from the R package `ecospat` `r as.character(packageVersion('ecospat'))` [@broennimann_ecospat_2018] with a grid resolution of 200.
We then determined the location in the environmental space of the plant-pollinator communities using the function `suprow` from `ade4`.
The normalised density at that particular location [which we calculated using the R package `raster` `r as.character(packageVersion('raster'))`; @hijmans_raster_2019] corresponds with the bioclimatic suitability. 
The result of all these steps is the generation of a value of environmental stress, which corresponds to one minus the bioclimatic suitability for a species of a particular location.  

```{r sensitivity-analysis-calculations}
loadd(sensitivity_species_name)
loadd(n_enough_occurrences)
n_occ_sensitivity_species <- n_thinned_occurrences %>%
  dplyr::filter(org_id == sensitivity_species_name$org_id[1]) %$% N

n_spp_not_enough_occ <- nrow(n_thinned_occurrences) - nrow(n_enough_occurrences$single_species)
```

We used a sensitivity analysis to determine the minimum number of occurrences that are necessary to have robust environmental stress estimations. 
For that we used the species with most occurrences available, *`r sensitivity_species_name$canonicalName`*, and calculated the mean absolute error of the bioclimatic suitability values obtained with `r numbers2words(config$n_subsamples)` subsamples from the `r format(n_occ_sensitivity_species, big.mark  =",")`  occurrences available from GBIF. 

## Data analysis

We then used a set of Bayesian multilevel models to evaluate the impact of environmental stress on species specialisation. 
Specifically, we use the normalised degree of species as our response variable; that is, the number of species it interacts with given the number of species in the opposite guild [@martin_gonzalez_centrality_2010]. 
The normalised degree was modelled using a logit link function and a binomial distribution in which the number of partner species a focal species interacts with is the number of successes, and the number of species in the opposite guild is the number of trials. 
We are aware that whether species interact or not is not a Bernoulli process as species interactions are not strictly independent from each other. 
However, the use of a binomial distribution allows us to account for the differences in species richness across communities indirectly. 
Importantly, results are qualitatively similar when we model species degree directly using a Poisson distribution and a logarithmic link function.

We evaluated four models to assess the relative importance of stress for each focal species.  
A first model, our baseline model, included five variables. 
<!-- fixed variables (population-level predictors) and two random variables (grouping levels). -->
The predictors in the baseline model were the environmental stress of the community for each focal species, the species' number of known possible partners in the community, and both the species guild (plant or a pollinator) and its interaction with environmental stress.
We included the number of known possible partners as a predictor in our models as it allows us to control for the effects of the environment on community composition, effectively accounting for species co-occurrence. 
We calculated this metric by determining the number of partners with which the species is known to interact in any other community and assessing how mant of these potential partners are present in the local community.
Controlling for the number of potential partners makes our model a particularly stringent test of our environmental-stress hypotheses because this variable could explain a large proportion of variance. 
Often, the potential and the actual number of partners is the same or very close to each other, especially for rare species present only in a few communities.

We allowed the intercept and slope of the stress-specialisation relationship to vary among species. 
This approach allowed us to investigate two questions. 
First, it allows us to inspect the extent to which environmental stress affects species similarly. 
Second, by investigating the correlation between the intercept and the slope as a model parameter, it allowed us to inspect the extent by which species with a small or large number of partner species respond to increasing levels of environmental stress. 
To account for unmeasured differences between communities, like sampling effort, sampling method, or diversity, we also allowed the model intercept to be different for each community in our study.
To facilitate model interpretation and convergence, we scaled all continuous variables to have a mean of zero and a unit variance. 

We compared this baseline model with three alternative models in which we removed one predictor at a time.
To quantify the difference between models, in terms of their expected out-of-sample performance, we use the Wanatabe-Akaike information criterion (WAIC).
All models were fitted under a Bayesian framework using the R package `brms` `r as.character(packageVersion('brms'))` [@burkner_brms_2017; @burkner_advanced_2018] as an interface for Stan [@carpenter_stan_2017]. 
For each model, we used four Markov chains of 4,000 iterations each; we used half of the iterations for warmup.
We used weakly informative priors for all model parameters. 
Specifically we used normal priors of mean zero and standard deviation ten for the population-level effects and the intercepts, a half-Cauchy prior with a location of zero and a scale of two for the standard deviations, and, when applicable, an LKJ-correlation prior with parameter \(\zeta = 1\) for the correlation matrix between group-level parameters.

# Results

```{r sensitivity-results}
loadd(approach_suitability)
threshold_type <- ifelse(approach_suitability == "independent_suitability", "single_species", "all_species")
```

After performing our sensitivity analysis, we found that, for a species, we need roughly `r readd(min_occurrences_factor)[[threshold_type]]` independent occurrences for each community for which we aim to estimate the environmental stress. 
This is the number of occurrences necessary to maintain the mean absolute error of bioclimatic suitability below `r config$min_suitability_error` (Fig. \@ref(fig:fig-sensitivity-analysis)). 
We therefore removed from further analyses `r n_spp_not_enough_occ` species for which we did not have enough occurrences to obtain robust estimates.
<!-- When inspecting the suitability values of the analysed species, we found that most communities included species for which habitat suitability was low and species for which it was high (Fig. \@ref(fig:fig-median-suitability)). -->

```{r fig-sensitivity-analysis, fig.width = width('single'), fig.height = 2.2, fig.cap = fig_sensitivity_analysis_legend}
loadd(sensitivity_species_name)
fig_sensitivity_analysis_legend <- 
  paste0("Sensitivity analysis of environmental stress error. ",
         "The number of independent occurrences retrieved from GBIF is inversely related to the error of bioclimatic suitability for our plant-pollinator networks.", 
         "The sensitivity analysis was performed by subsampling occurrences of", 
         " *", sensitivity_species_name$canonicalName, "* ", 
         "the species in our dataset with the largest number of occurrences in GBIF, which was recorded in two of our communities.")

readd(fig_sensitivity_analysis)
```

```{r model-results-calculations}
loadd(bayesian_r2_baseline)
loadd(parameter_posterior_summary)
loadd(model_ranking)

random_cor <- parameter_posterior_summary %>%
  dplyr::filter(parameter == "cor_org_id__Intercept__scaled_suitability")

sd_loc_id <- parameter_posterior_summary %>% 
  dplyr::filter(stringr::str_detect(parameter, "sd_loc_id__Intercept"))

sd_org_id <- parameter_posterior_summary %>% 
  dplyr::filter(stringr::str_detect(parameter, "sd_org_id__Intercept"))

sd_org_id_suitability <- parameter_posterior_summary %>% 
  dplyr::filter(stringr::str_detect(parameter, "sd_org_id__scaled_suitability"))

waic_comp_suitability <- model_ranking$waic$formula_no_suitability$pointwise - model_ranking$waic$formula_base$pointwise 

waic_diff_suitability <- sum(waic_comp_suitability[, "waic"])

waic_se_diff_suitability <- sqrt(nrow(waic_comp_suitability) * var(waic_comp_suitability[, "waic"]))
```

Our models performed relatively well.
The Bayesian R^2^ for our baseline model was `r round(bayesian_r2_baseline[1], digits = 2)`, which indicates our models were able to capture a large proportion of the variability on the data. 
Overall, we found that environmental stress does not have a consistent effect across species. 
Indeed, when looking at the fixed effects, stress has virtually no relationship with the normalised degree—our metric of specialisation (Figure \@ref(fig:fig-conditional-effects)a). 
However, environmental stress was still an important predictor in our model. 
The difference in WAIC between our baseline model and the model that did not include environmental stress was `r round(waic_diff_suitability)` ± `r round(waic_se_diff_suitability)` (Table \@ref(tab:tab-model-comparison)).
This apparent discrepancy can be explained by the variability of the specialisation-stress relationship across species.

```{r fig-conditional-effects, fig.width = width('single'), fig.height = 2.2*1.5, fig.cap = fig_conditional_effects_legend}
loadd(median_trials)

median_n_pla <- median_trials %>%
  dplyr::filter(guild == "pla_id") %$% 
  n_opposite_guild

median_n_ani <- median_trials %>%
  dplyr::filter(guild == "ani_id") %$% 
  n_opposite_guild

fig_conditional_effects_legend <- 
  paste0("Conditional effects of predictors in our baseline model. The shown values are based on predictions for a hypothetical community with ",
         median_n_pla, " plants and ", 
         median_n_ani, " pollinators",
         ". These values correspond to the median number of species in each guild across communities. In each panel, we condition on the mean value of the other predictor in the model. We indicate mean values for each predictor with a vertical dashed line. For model fitting, we scaled all predictors to have a mean of zero and unit variance; however, here we show the unscaled predictors to facilitate interpretation. To illustrate the uncertainty around the fitted estimates, we plot the fits of 100 independent draws from the posterior distribution. The thick lines indicate the mean values of the response distribution. As there was no interaction between the guild and the number of possible interactions, we only show the conditional effect of pollinators.")

readd(fig_conditional_effects)
```

```{r tab-model-comparison}
model_ranking$waic %>%
  purrr::map(~.$estimates["waic", ]) %>% 
  purrr::map_df(~tibble::tibble(estimate = .[1], se = .[2]), .id = "formula") %>%
  dplyr::arrange(estimate) %>%
  dplyr::filter(formula != "formula_full") %>%
  dplyr::mutate(formula = translate_model_formula(formula, "long-abv"), 
                formula = kableExtra::cell_spec(formula, format = "latex", 
                                                bold =nchar(formula) == max(nchar(formula)))) %>% 
  knitr::kable("latex", 
               booktabs = TRUE, 
               digits = c(0,0,0), 
               format.args = list(big.mark = ","), 
               col.names = c("predictors", "WAIC", "SE"),
               escape = FALSE,
               linesep = "",
               caption = "Comparison in out of sample predictive power of the baseline model (bold) and their alternatives. We rank models by their expected log predictive density based on their Wanatabe-Akaike information criterion (WAIC).") %>%
  kableExtra::kable_styling(font_size = 8) #%>%
```

For some species, there is a strong negative relationship between stress and specialisation, while for others, there is a strong positive relationship (Figure \@ref(fig:fig-random-effects)a). 
Interestingly, the slope of this relationship correlates with the species' intercept in the model (Figure \@ref(fig:fig-random-effects)b and c).
Recall that the model estimates the intercept at the mean value for stress across communities (`r round(readd(mean_parameter_values)$suitability, 2)`).
The mean correlation coefficient was `r round(random_cor$Estimate, 2)` [`r round(random_cor$Q2.5, 2)`, `r round(random_cor$Q97.5, 2)`].
Therefore, the slope of the stress-specialisation relationship was more likely to be positive for species with a large number of partners under average stress conditions (and negative for species with a smaller number of partners).
Extrapolating to no-stress conditions: species that would interact with a small number of partners under no stress are more likely to interact with more partners as stress increases, whereas those that would interact with a large number of partners are more likely to interact with less.

```{r fig-random-effects, fig.width = width('double'), fig.height = 2.2*2, fig.cap = fig_random_effects_legend, out.width = "6.81in", out.height = "4.4in", out.extra="center"}
loadd(random_species_draws)

n_random_species <- random_species_draws %>%
  dplyr::group_by(guild) %>%
  dplyr::summarise(n_sp = dplyr::n_distinct(org_id))

n_random_pla <- n_random_species %>%
  dplyr::filter(guild == "pla_id") %$%
  n_sp

n_random_ani <- n_random_species %>%
  dplyr::filter(guild == "ani_id") %$%
  n_sp

fig_random_effects_legend <- 
  paste0("Species-level effects of environmental stress (a) Conditional effect of stress for individual species. Each line corresponds to the median relationship for each species. Although we included in the analysis of all species that are present in two or core communities, to facilitate visualisation here, we show only species for which there is suitability information in at least six communities (", 
         n_random_pla, " plants and ", 
         n_random_ani, " pollinators). As in the previous figure, fitted values assume a hypothetical community of median size. In each panel, we highlight two species for which the relationship between environmental suitability and the normalised degree was particularly strong. ", 
         "(b and c) The correlation between the species' intercept and the species' slope of suitability was negative. The species' intercept can be interpreted as the relative difference between the number of partners a species has under mean levels of environmental stress and the mean number of partners across all species. Positive values of species' slope indicate a positive relationship between stress and the number of partners and vice-versa.")

readd(fig_random_effects)
```

As expected, we found a strong and positive relationship between the number of possible interactions and the number of realised interactions in the community. 
There was also a large difference of WAIC between the model that included this predictor and that that excluded it. 
This result indicates that the availability of potential partners—this is, community composition—accounts for a large proportion of the variability in species degree. 
Importantly, our findings relating to the variability of the stress-specialisation relationship were qualitatively unchanged, whether we included this variable or not. 

The standard deviation (in the parameters scale) of the community intercepts was `r round(sd_loc_id$Estimate, 2)` [`r round(sd_loc_id$Q2.5, 2)`, `r round(sd_loc_id$Q97.5, 2)`] which indicates the importance of the local context when determining specialisation. 
The standard deviation of the species intercept was `r round(sd_org_id$Estimate, 2)` [`r round(sd_org_id$Q2.5, 2)`, `r round(sd_org_id$Q97.5, 2)`], and that of the species' stress slope was `r round(sd_org_id_suitability$Estimate, 2)` [`r round(sd_org_id_suitability$Q2.5, 2)`, `r round(sd_org_id_suitability$Q97.5, 2)`] (95% credible intervals shown within square brakets). 

# Discussion 

We set out to explore whether and how environmental stress can systematically affect specialisation. 
After accounting for the pool of potential partners, we found that environmental conditions contribute to determining whether a species is a generalist or a specialist *in their community*. 
We also found that the particular effect of the environment is strongly dependent on the species.
Based on existing literature, we proposed two alternative hypotheses of how environmental stress may affect species' specialisation, and we found evidence for both of them.
Species with a large number of partners in low-stress communities were more likely to have a negative relationship and hence reduce the number of partners as stress increases. 
Contrastingly, species in our datasets with a small number of partners in low-stress communities were more likely to have a larger number of partners in more stressful communities.
In summary, environmental stress pushes species that are flexible enough to change their interaction partners towards intermediate levels of specialisation, a so-called "regression towards the mean".

Our results suggest that changes in community composition are indeed the primary channel through which the environment determines changes interaction probability. 
However, they also show that, for a large number of species, the environment may also play a substantial role in determining their level of specialisation. 
Previous research has recognised that environmental factors may help explain the changes in network structure along environmental gradients that cannot be explained by community composition [@tylianakis_habitat_2007].
However, how these two factors were linked had been elusive so far [@gravel_bringing_2018]. 
We believe that part of this difficulty could have arisen because species, and ultimately network structure, can respond in multiple, and contrasting, ways depending on the particular bioclimatic variable examined (e.g. temperature or precipitation).
Using stress to summarise the effect on species of multiple environmental gradients allowed us to detect a clear signal of the environment in species' interaction patterns.

Although both niche and neutral processes are relevant in determining species interactions, our model suggests that niche processes may be the predominant mechanism through which the environment *systematically* affects specialisation. 
First, it is unlikely that environmental stress correlates to local species abundances [@pearce_practical_2001; @sagarin_moving_2006]. 
Second, even if there is a relationship between stress and abundances, a particular environmental gradient could have a positive effect on the abundance of some species and a negative effect on others. 
Indeed, we find that within a community there is a wide range of stress values, even for the relatively limited number of species we were able to include in our analysis.

Recent research suggests that species are continuously changing their interaction partners wherever environmental conditions change in space or time [@raimundo_adaptive_2018].
So far it appears that this rewiring is primarily driven by generalist species [@ponisio_opportunistic_2017; @burkle_plant-pollinator_2013], presumably because generalist species are less sensitive to trait matching of their interaction partners [@caradonna_interaction_2017].
Our results add two important nuances to these findings. 
First, because "generalists" seem to focus on a smaller number of partners as environmental conditions deteriorate, we show that trait matching might still play a role in determining the interactions of generalist species. 
Second, and most importantly, our results suggest that only a small proportion of species are "true generalists" or "true specialists" this is, species that interact with a large or small number of partners regardless of the environmental stress, respectively. 
This pattern implies that rewiring is not exclusive of species with a large number of partners. 
Instead, at least a fraction of the species that appear to be specialist *in their communities* might be as flexible, if not more, than those with a large number of partners, effectively behaving as facultative generalists in the face of environmental change. 
These flexible "specialists" might therefore have a more significant role in network persistence than previously expected.

In our model, we can roughly divide species between true specialists, true generalists, and flexible species. 
However, there is a fourth group that remained invisible to our model but has important implications for network persistence and stability.
Species that can vary their interaction partners flexibly and their role in the network are more likely to persist in their community as environmental conditions vary [@gaiarsa_interaction_2019].
We propose this fourth group is composed of true specialists that are constrained to interact with partners of high trait-matching and therefore were not likely to be found in more than one community. 
If species that are not flexible are unlikely to persist over temporal or spatial environmental gradients, we can expect specialised communities that are highly constrained by trait-matching [like some plant-hummingbird networks; @vizentin-bugoni_processes_2014; @maruyama_morphological_2014] to be far more vulnerable to increased climate change-induced environmental stress and habitat degradation than communities where role and interaction flexibility are more prevalent.

Similarly, if the patterns we see in our models have also played a role during the evolutionary history of pollination communities, our results also help explain why only a small fraction of plant-pollinator interactions show a strong signature of deep co-evolutionary history [@hutchinson_cophylogenetic_2017].
The increases in the stress that species are predicted to experience due to rapid environmental change might further erode the co-evolutionary history of specialist species.
Communities as a whole might be in a trajectory of even more diffuse co-evolution.
For specialists, at least, the longer-term benefits of being able to interact with multiple partners might be more important than the shorter-term benefits of interacting with partners of high trait matching. 

The structural implications of the "regression towards the mean" that environmental stress promotes are less clear. 
However, it is plausible to expect that nestedness, and therefore network stability, might be reduced in the face of rapid environmental change.
Determining exactly how the changes in degree caused by environmental stress reflect on systematic changes in network structure would be an interesting avenue of research. 
Answering this question would require expanding our suitability analysis to all species in the community and compare the degree distribution of networks along a gradient of stress for the community as a whole. 

In conclusion, we show that the environment can affect the specialisation level of plants and pollinators in systematic ways beyond community composition. 
Species that are inflexible with their interaction partners are unlikely to persist under more stressful environmental conditions. 
However, we show that many species are flexible in regards to their specialisation levels and therefore are not inherently generalists or specialists. 
Instead, the species' level of specialisation/generalisation should be considered on a relative scale depending on where they are found and the environmental conditions at that location. 

# Acknowledgements

`r paste(drake::readd(acknowledgements), collapse = "")`

# References

